--- build.xml	2004-04-30 20:02:34.000000000 -0500
+++ build.xml.patched	2004-05-08 14:16:02.487920896 -0500
@@ -19,6 +19,13 @@
   on date February 16 2004, time 2310-->
 
 <project default="jar" name="commons-net" basedir=".">
+  <path id="compile.classpath">
+  <pathelement location="target/classes"/>
+  <pathelement location="target/test-classes"/>
+  <pathelement location="target/test-reports"/>
+  <pathelement location="${oro.jar}"/>
+  <pathelement location="${junit.jar}"/>
+  </path>
   <property name="defaulttargetdir" value="target">
   </property>
   <property name="libdir" value="target/lib">
@@ -43,7 +50,7 @@
       </equals>
     </condition>
   </target>
-  <target name="compile" description="o Compile the code" depends="get-deps">
+  <target name="compile" description="o Compile the code">
     <mkdir dir="${classesdir}">
     </mkdir>
     <javac destdir="${classesdir}" deprecation="true" debug="true" optimize="false" excludes="**/package.html">
@@ -51,15 +58,10 @@
         <pathelement location="src/java">
         </pathelement>
       </src>
-      <classpath>
-        <fileset dir="${libdir}">
-          <include name="*.jar">
-          </include>
-        </fileset>
-      </classpath>
+	<classpath refid="compile.classpath"/>
     </javac>
   </target>
-  <target name="jar" description="o Create the jar" depends="compile,test">
+  <target name="jar" description="o Create the jar" depends="compile">
     <jar jarfile="target/${final.name}.jar" excludes="**/package.html" basedir="${classesdir}">
     </jar>
   </target>
@@ -143,30 +145,13 @@
     <property name="title" value="Jakarta Commons/Net 1.2.0-dev API">
     </property>
     <javadoc use="true" private="true" destdir="${javadocdir}" author="true" version="true" sourcepath="src/java" packagenames="org.apache.commons.net.*">
-      <classpath>
-        <fileset dir="${libdir}">
-          <include name="*.jar">
-          </include>
-        </fileset>
-        <pathelement location="target/${final.name}.jar">
-        </pathelement>
-      </classpath>
+	<classpath refid="compile.classpath"/>
     </javadoc>
   </target>
-  <target name="get-deps" unless="noget" depends="init">
-    <get dest="${libdir}/oro-2.0.8.jar" usetimestamp="true" ignoreerrors="true" src="http://www.ibiblio.org/maven/oro/jars/oro-2.0.8.jar">
-    </get>
-    <get dest="${libdir}/junit-3.8.1.jar" usetimestamp="true" ignoreerrors="true" src="http://www.ibiblio.org/maven/junit/jars/junit-3.8.1.jar">
-    </get>
-    <get dest="${libdir}/ant-1.5.jar" usetimestamp="true" ignoreerrors="true" src="http://www.ibiblio.org/maven/ant/jars/ant-1.5.jar">
-    </get>
-    <get dest="${libdir}/ant-optional-1.5.jar" usetimestamp="true" ignoreerrors="true" src="http://www.ibiblio.org/maven/ant/jars/ant-optional-1.5.jar">
-    </get>
-  </target>
   <target name="install-maven">
     <get dest="${user.home}/maven-install-latest.jar" usetimestamp="true" src="${repo}/maven/maven-install-latest.jar">
     </get>
     <unjar dest="${maven.home}" src="${user.home}/maven-install-latest.jar">
     </unjar>
   </target>
-</project>
\ No newline at end of file
+</project>
