--- Azureus_3.0.3.4_source.orig/com/aelitis/azureus/ui/swt/views/skin/TorrentListViewsUtils.java	2007-10-01 14:48:00.000000000 +0200
+++ Azureus_3.0.3.4_source/com/aelitis/azureus/ui/swt/views/skin/TorrentListViewsUtils.java	2007-11-02 16:37:00.000000000 +0100
@@ -64,7 +64,6 @@
 import com.aelitis.azureus.ui.swt.views.TorrentListViewListener;
 import com.aelitis.azureus.util.AdManager;
 import com.aelitis.azureus.util.Constants;
-import com.aelitis.azureus.util.win32.Win32Utils;
 
 import org.gudy.azureus2.plugins.PluginInterface;
 import org.gudy.azureus2.plugins.PluginManager;
@@ -602,17 +601,6 @@
 	 * @param string
 	 */
 	private static boolean runInMediaPlayer(String mediaFile) {
-		if (Constants.isWindows) {
-			String wmpEXE = Win32Utils.getWMP();
-			if (new File(wmpEXE).exists()) {
-				try {
-					Runtime.getRuntime().exec(wmpEXE + " \"" + mediaFile + "\"");
-					return true;
-				} catch (IOException e) {
-					Debug.out("error playing " + mediaFile + " via WMP " + mediaFile, e);
-				}
-			}
-		}
 		return false;
 	}
 
