--- Azureus_3.0.3.4_source.orig/org/gudy/azureus2/platform/PlatformManagerFactory.java	2007-01-12 13:39:24.000000000 +0100
+++ Azureus_3.0.3.4_source/org/gudy/azureus2/platform/PlatformManagerFactory.java	2007-11-02 16:51:05.000000000 +0100
@@ -46,23 +46,7 @@
 			if ( platform_manager == null ){
 										  
 				try{
-					if ( getPlatformType() == PlatformManager.PT_WINDOWS ){
-						
-						platform_manager = org.gudy.azureus2.platform.win32.PlatformManagerImpl.getSingleton();
-						
-					}else if( getPlatformType() == PlatformManager.PT_MACOSX ){
-						
-	                    platform_manager = org.gudy.azureus2.platform.macosx.PlatformManagerImpl.getSingleton();
-	                    
-					}else if( getPlatformType() == PlatformManager.PT_UNIX ){
-						
             platform_manager = org.gudy.azureus2.platform.unix.PlatformManagerImpl.getSingleton();
-
-					}
-				}catch( PlatformManagerException e ){
-					
-						// exception will already have been logged
-					
 				}catch( Throwable e ){
 					
 					Debug.printStackTrace(e);
