diff -ur work.old/org/gudy/azureus2/platform/PlatformManagerFactory.java work/org/gudy/azureus2/platform/PlatformManagerFactory.java
--- work.old/org/gudy/azureus2/platform/PlatformManagerFactory.java	2008-04-20 15:53:14.000000000 +0300
+++ work/org/gudy/azureus2/platform/PlatformManagerFactory.java	2008-04-20 15:53:31.000000000 +0300
@@ -52,7 +52,7 @@
 						
 					}else if( getPlatformType() == PlatformManager.PT_MACOSX ){
 						
-	                    platform_manager = org.gudy.azureus2.platform.macosx.PlatformManagerImpl.getSingleton();
+	                    //platform_manager = org.gudy.azureus2.platform.macosx.PlatformManagerImpl.getSingleton();
 	                    
 					}else if( getPlatformType() == PlatformManager.PT_UNIX ){
 						
