diff -ur maven-script-2.0/maven-script-ant/src/main/java/org/apache/maven/script/ant/AntMojoComponentFactory.java new/maven-script-ant/src/main/java/org/apache/maven/script/ant/AntMojoComponentFactory.java
--- maven-script-2.0/maven-script-ant/src/main/java/org/apache/maven/script/ant/AntMojoComponentFactory.java	2007-05-08 13:26:13.000000000 +0200
+++ new/maven-script-ant/src/main/java/org/apache/maven/script/ant/AntMojoComponentFactory.java	2007-05-25 19:24:47.894856172 +0200
@@ -30,7 +30,7 @@
     extends AntComponentFactory
 {
 
-    public Object newInstance( ComponentDescriptor descriptor, ClassRealm realm, PlexusContainer container )
+    public Object newInstance( ComponentDescriptor descriptor, org.codehaus.plexus.classworlds.realm.ClassRealm  realm, PlexusContainer container )
         throws ComponentInstantiationException
     {
         return new AntMojoWrapper( (AntScriptInvoker) super.newInstance( descriptor, realm, container ) );
Seulement dans new/maven-script-ant/src/main/resources/META-INF: maven
Seulement dans new/maven-script-beanshell/src/main: resources
Seulement dans new: src
