--- plugins/org.eclipse.pde.api.tools.ui/src/org/eclipse/pde/api/tools/ui/internal/wizards/ApiProfileWizardPage.java.orig	2008-04-01 11:24:25.000000000 +0200
+++ plugins/org.eclipse.pde.api.tools.ui/src/org/eclipse/pde/api/tools/ui/internal/wizards/ApiProfileWizardPage.java	2008-04-01 11:25:43.000000000 +0200
@@ -282,7 +282,7 @@
 		}
 		Collections.sort(items, new Comparator() {
 			public int compare(Object o1, Object o2) {
-				return ((String)o1).compareTo(o2);
+				return ((String)o1).compareTo((String)o2);
 			}
 		});		
 				
