--- plugins/org.eclipse.pde.api.tools.ui/src/org/eclipse/pde/api/tools/ui/internal/wizards/ApiProfileWizardPage.java.orig	2008-04-01 11:24:25.000000000 +0200
+++ plugins/org.eclipse.pde.api.tools.ui/src/org/eclipse/pde/api/tools/ui/internal/wizards/ApiProfileWizardPage.java	2008-04-01 11:25:43.000000000 +0200
@@ -282,7 +282,7 @@
 		}
 		Collections.sort(items, new Comparator() {
 			public int compare(Object o1, Object o2) {
-				return ((String)o1).compareTo(o2);
+				return ((String)o1).compareTo((String)o2);
 			}
 		});		
 				
--- plugins/org.eclipse.pde.api.tools/src_ant/org/eclipse/pde/api/tools/internal/tasks/DeltaReportConversionTask.java.orig	2008-05-05 21:35:49.000000000 +0200
+++ plugins/org.eclipse.pde.api.tools/src_ant/org/eclipse/pde/api/tools/internal/tasks/DeltaReportConversionTask.java	2008-05-05 21:36:16.000000000 +0200
@@ -243,7 +243,7 @@
 			public int compare(Object o1, Object o2) {
 				Map.Entry entry1 = (Map.Entry) o1;
 				Map.Entry entry2 = (Map.Entry) o2;
-				return ((String) entry1.getKey()).compareTo(entry2.getKey());
+				return ((String)entry1.getKey()).compareTo((String)entry2.getKey());
 			}
 		});
 		for (Iterator iterator = allEntries.iterator(); iterator.hasNext(); ) {
--- plugins/org.eclipse.pde.api.tools/src_ant/org/eclipse/pde/api/tools/internal/tasks/APIToolsVerificationTask.java.orig	2008-06-11 17:16:18.000000000 +0200
+++ plugins/org.eclipse.pde.api.tools/src_ant/org/eclipse/pde/api/tools/internal/tasks/APIToolsVerificationTask.java	2008-06-11 17:16:39.000000000 +0200
@@ -656,7 +656,7 @@
 			public int compare(Object o1, Object o2) {
 				Map.Entry entry1 = (Map.Entry) o1;
 				Map.Entry entry2 = (Map.Entry) o2;
-				return ((String) entry1.getKey()).compareTo(entry2.getKey());
+				return ((String)entry1.getKey()).compareTo((String)entry2.getKey());
 			}
 		});
 		Summary[] summaries = new Summary[size];
