--- org/gudy/azureus2/pluginsimpl/local/PluginInitializer.java.orig	2006-01-11 09:05:48.000000000 -0800
+++ org/gudy/azureus2/pluginsimpl/local/PluginInitializer.java	2006-01-25 11:51:54.000000000 -0800
@@ -133,11 +133,6 @@
    					"azbpcorepatcher", 
    					"CorePatcher",
 					"true" },
-	   		{	 PluginManagerDefaults.PID_PLATFORM_CHECKER, 
-   					"org.gudy.azureus2.platform.win32.PlatformManagerUpdateChecker", 
-   					"azplatform2", 
-   					"azplatform2",
-					"true" },
 	   		{	 PluginManagerDefaults.PID_JPC, 
 					"com.aelitis.azureus.plugins.jpc.JPCPlugin", 
 					"azjpc", 
