--- org/gudy/azureus2/ui/swt/mainwindow/MainMenu.java~	2006-02-09 20:09:11.000000000 -0500
+++ org/gudy/azureus2/ui/swt/mainwindow/MainMenu.java	2006-02-09 20:09:17.000000000 -0500
@@ -171,17 +171,6 @@
       if(!Constants.isOSX) {
         new MenuItem(fileMenu, SWT.SEPARATOR);
 
-        MenuItem file_restart = new MenuItem(fileMenu, SWT.NULL);
-        Messages.setLanguageText(file_restart, "MainWindow.menu.file.restart"); //$NON-NLS-1$
-
-        file_restart.addListener(SWT.Selection, new Listener() {
-
-        public void handleEvent(Event event) {
-
-            MainWindow.getWindow().dispose(true,false);
-         }
-        });
-
         final MenuItem file_exit = new MenuItem(fileMenu, SWT.NULL);
         if(!COConfigurationManager.getBooleanParameter("Enable System Tray") || !COConfigurationManager.getBooleanParameter("Close To Tray")) {
             KeyBindings.setAccelerator(file_exit, "MainWindow.menu.file.exit");
