--- DBusConnection.java	2006-11-13 02:48:18.000000000 +0200
+++ libdbus-java-1.12/org/freedesktop/dbus/DBusConnection.java	2006-12-08 00:14:16.000000000 +0200
@@ -379,7 +379,6 @@
    private native int dbus_call_method(int connid, String busname, String objectpath, String type, String name, int flags, Object... params);
    private native int dbus_reply_to_call(int connid, String destination, String type, String objectpath, String name, long replyserial, Object... params);
    static {
-      System.loadLibrary("dbus-1");
       System.loadLibrary("dbus-java");
    }
    private static final Map<Object,DBusConnection> conn = new HashMap<Object,DBusConnection>();
