diff -ru jakarta-turbine-jcs/src/java/org/apache/jcs/engine/CacheEventQueue.java jakarta-turbine-jcs-orig/src/java/org/apache/jcs/engine/CacheEventQueue.java
--- jakarta-turbine-jcs/src/java/org/apache/jcs/engine/CacheEventQueue.java	2005-06-03 22:01:54.000000000 -0400
+++ jakarta-turbine-jcs-orig/src/java/org/apache/jcs/engine/CacheEventQueue.java	2005-08-03 23:22:17.000000000 -0400
@@ -460,7 +460,7 @@
     {
         Node next = null;
 
-        AbstractCacheEvent event = null;
+        CacheEventQueue.AbstractCacheEvent event = null;
     }
 
     /**
diff -ru jakarta-turbine-jcs/src/java/org/apache/jcs/engine/control/event/ElementEventQueue.java jakarta-turbine-jcs-orig/src/java/org/apache/jcs/engine/control/event/ElementEventQueue.java
--- jakarta-turbine-jcs/src/java/org/apache/jcs/engine/control/event/ElementEventQueue.java	2005-06-03 22:01:59.000000000 -0400
+++ jakarta-turbine-jcs-orig/src/java/org/apache/jcs/engine/control/event/ElementEventQueue.java	2005-08-03 23:17:25.000000000 -0400
@@ -210,7 +210,7 @@
     {
         Node next = null;
 
-        AbstractElementEventRunner event = null;
+        ElementEventQueue.AbstractElementEventRunner event = null;
     }
 
     /**
