--- /usr/local/tmp/portage/dev-java/jrefactory-2.9.19/work/src/org/acm/seguin/findbugs/FindBugsFrame.java	2007-02-18 12:42:29.258645825 +0100
+++ FindBugsFrame.java	2007-02-18 12:43:05.385954672 +0100
@@ -151,13 +151,15 @@
          IDEPlugin.log(IDEInterface.DEBUG, this, "sourceFileName=" + sourceFileName);
 */
       } else {
-         return super.viewSource(project, analysisRun, srcLine); //, selected);
+      //   return super.viewSource(project, analysisRun, srcLine); //, selected);
+        return true;
       }
 
       IDEPlugin.log(IDEInterface.DEBUG, this, "sourceFileName=" + sourceFileName);
       if (sourceFileName == null || sourceFileName.equals("<Unknown>")) {
          getLogger().logMessage(ConsoleLogger.WARNING, "No source file for class " + srcLine.getClassName());
-         return super.viewSource(project, analysisRun, srcLine); //, selected);
+         //return super.viewSource(project, analysisRun, srcLine); //, selected);
+	 return true;
       }
 
       SourceFile sourceFile = sourceFinder.findSourceFile(packageName, sourceFileName);
@@ -193,8 +195,9 @@
             IDEPlugin.moveCaretPosition(view, buffer, end);
          }
       }
-
-      return super.viewSource(project, analysisRun, srcLine); //, selected);
+	
+//      return super.viewSource(project, analysisRun, srcLine); //, selected);
+      return true;
    }
 
 
