Index: Game/Core.java
===================================================================
--- Game/Core.java	(revision 53)
+++ Game/Core.java	(working copy)
@@ -4,6 +4,7 @@
 import java.awt.MediaTracker;
 import java.awt.Toolkit;
 import java.io.UnsupportedEncodingException;
+import java.io.File;
 import java.net.URL;
 import java.net.URLDecoder;
 import java.util.ArrayList;
@@ -79,32 +80,9 @@
 	 * @throws LemmException
 	 */
 	public static void init(final JFrame frame, final boolean isWebstartApp) throws LemmException  {
-		// get ini path
-		if (isWebstartApp) {
-			programPropsFileStr = ToolBox.exchangeSeparators(System.getProperty("user.home"));
-			programPropsFileStr = ToolBox.addSeparator(programPropsFileStr);
-		} else {
-			String s = frame.getClass().getName().replace('.','/') + ".class";
-			URL url = frame.getClass().getClassLoader().getResource(s);
-			int pos;
-			try {
-				programPropsFileStr = URLDecoder.decode(url.getPath(),"UTF-8");
-			} catch (UnsupportedEncodingException ex) {};
-			// special handling for JAR
-			if (( (pos=programPropsFileStr.toLowerCase().indexOf("file:")) != -1))
-				programPropsFileStr = programPropsFileStr.substring(pos+5);
-			if ( (pos=programPropsFileStr.toLowerCase().indexOf(s.toLowerCase())) != -1)
-				programPropsFileStr = programPropsFileStr.substring(0,pos);
-
-			/** @todo doesn't work if JAR is renamed...
-			 *  Maybe it would be a better idea to search only for ".JAR" and then
-			 *  for the first path separator...
-			 */
-
-			s = (frame.getClass().getName().replace('.','/') + ".jar").toLowerCase();
-			if ( (pos=programPropsFileStr.toLowerCase().indexOf(s)) != -1)
-				programPropsFileStr = programPropsFileStr.substring(0,pos);
-		}
+		String userDirStr = System.getProperty("user.home") + File.separator + ".lemmini";
+		userDirStr = ToolBox.exchangeSeparators(userDirStr);
+		programPropsFileStr = ToolBox.addSeparator(new String(userDirStr));
 		programPropsFileStr += INI_NAME;
 		// read main ini file
 		programProps = new Props();
@@ -118,7 +96,7 @@
 
 		scale = Core.programProps.get("scale",1.0);
 
-		resourcePath = programProps.get("resourcePath", "");
+		resourcePath = programProps.get("resourcePath", userDirStr);
 		String sourcePath = programProps.get("sourcePath", "");
 		String rev = programProps.get("revision", "");
 		GameController.setMusicOn(programProps.get("music", false));
