--- Makefile.rules.orig	2008-08-21 10:54:42.000000000 +0300
+++ Makefile.rules	2008-08-21 10:55:27.000000000 +0300
@@ -398,11 +398,11 @@
 endif
 
 # By default, strip symbol information from executable
-ifndef KEEP_SYMBOLS
-  Strip := $(PLATFORMSTRIPOPTS)
-  StripWarnMsg := "(without symbols)"
-  Install.StripFlag += -s
-endif
+#ifndef KEEP_SYMBOLS
+#  Strip := $(PLATFORMSTRIPOPTS)
+#  StripWarnMsg := "(without symbols)"
+#  Install.StripFlag += -s
+#endif
 
 # Adjust linker flags for building an executable
 ifdef TOOLNAME
