--- build.xml	2008-10-04 13:50:55.000000000 +0300
+++ build.xml.fixed	2008-10-04 15:19:14.000000000 +0300
@@ -78,7 +78,7 @@
   <!-- TEST -->
   <!-- ================================================================== -->
 
-  <target name="test" depends="clean,init,build" description="Runs unit tests">
+  <target name="test" description="Runs unit tests">
 
 	<mkdir dir="${junit.classes.dir}"/>
 	<mkdir dir="${dist.dir}/report/tests"/>
@@ -239,14 +239,6 @@
 	  debug="${debug}">
       <classpath refid="classpath"/>
     </javac>
-  	<javac encoding="${encoding}"
-      srcdir="${src.test.dir}"
-      includes="de/danielnaber/languagetool/rules/patterns/PatternRuleTest.java"
-      destdir="${build.classes}"
-      target="${target}"
-	  debug="${debug}">
-      <classpath refid="classpath"/>
-    </javac>
 
   </target>
 
@@ -267,7 +259,7 @@
 		excludes="**/*.java">
 		<manifest>
 			<attribute name="Main-Class" value="de.danielnaber.languagetool.Main" />
-			<attribute name="Class-Path" value="./ ./rules ./resource ./${stempelator.lib}  ./jdic.jar ./jaminid.jar ./jWordSplitter.jar" />
+			<attribute name="Class-Path" value="./ ../rules ../resource ./${stempelator.lib}  ./jdic.jar ./jaminid.jar ./jWordSplitter.jar" />
 			<attribute name="Implementation-Version" value="${version}" />
 			<attribute name="Implementation-Date" value="${date}" />
 		</manifest>
@@ -279,7 +271,7 @@
 		includes="de/danielnaber/languagetool/gui/*.class">
 		<manifest>
 			<attribute name="Main-Class" value="de.danielnaber.languagetool.gui.Main" />
-			<attribute name="Class-Path" value="./LanguageTool.jar ./ ./resource ./rules ./${stempelator.lib}  ./jdic.jar ./jaminid.jar ./jWordSplitter.jar" />
+			<attribute name="Class-Path" value="./LanguageTool.jar ../ ../resource ./rules ./${stempelator.lib}  ./jdic.jar ./jaminid.jar ./jWordSplitter.jar" />
 			<attribute name="Implementation-Version" value="${version}" />
 			<attribute name="Implementation-Date" value="${date}" />
 		</manifest>
