diff -Nur languagetool-20081004/src/test/de/danielnaber/languagetool/ValidateXMLTest.java languagetool-20081004_patched/src/test/de/danielnaber/languagetool/ValidateXMLTest.java
--- languagetool-20081004/src/test/de/danielnaber/languagetool/ValidateXMLTest.java	2008-04-06 00:26:06.000000000 +0300
+++ languagetool-20081004_patched/src/test/de/danielnaber/languagetool/ValidateXMLTest.java	2008-10-05 00:04:00.000000000 +0300
@@ -28,14 +28,14 @@
     XMLValidator validator = new XMLValidator();
     for (int i = 0; i < Language.LANGUAGES.length; i++) {
       Language lang = Language.LANGUAGES[i];
-      String grammarFile = "/rules/" + lang.getShortName() + "/grammar.xml";
-      validator.validate(grammarFile, "/rules/rules.dtd", "rules");
+      String grammarFile = "src/rules/" + lang.getShortName() + "/grammar.xml";
+      validator.validate(grammarFile, "src/rules/rules.dtd", "rules");
     }
   }
 
   public void testFalseFriendsXML() throws IOException {
     XMLValidator validator = new XMLValidator();
-    validator.validate("/rules/false-friends.xml", "/rules/false-friends.dtd", "rules");
+    validator.validate("src/rules/false-friends.xml", "src/rules/false-friends.dtd", "rules");
   }
 
   public void testDisambiguationRuleFile() throws IOException {
@@ -43,11 +43,11 @@
     //for (int i = 0; i < Language.LANGUAGES.length; i++) {
     //  Language lang = Language.LANGUAGES[i];
     Language lang = Language.FRENCH;
-    String grammarFile = "/resource/" + lang.getShortName() + "/disambiguation.xml";
-    validator.validate(grammarFile, "/resource/disambiguation.dtd", "rules");
+    String grammarFile = "src/resource/" + lang.getShortName() + "/disambiguation.xml";
+    validator.validate(grammarFile, "src/resource/disambiguation.dtd", "rules");
     lang = Language.ENGLISH;
-    grammarFile = "/resource/" + lang.getShortName() + "/disambiguation.xml";
-    validator.validate(grammarFile, "/resource/disambiguation.dtd", "rules");
+    grammarFile = "src/resource/" + lang.getShortName() + "/disambiguation.xml";
+    validator.validate(grammarFile, "src/resource/disambiguation.dtd", "rules");
     // }
   }
 
