--- build.xml	2009-04-26 01:01:33.000000000 +0300
+++ build.xml.fixed	2009-04-26 22:28:05.000000000 +0300
@@ -18,10 +18,15 @@
     <pathelement location="${ext.jaminid.lib}"/>
     <pathelement location="${ext.jwordsplitter.lib}"/>
   	<!-- -->
-    <pathelement location="${ext.ooo.juh.lib}"/>
-    <pathelement location="${ext.ooo.unoil.lib}"/>
-    <pathelement location="${ext.ooo.ridl.lib}"/>
-    <pathelement location="${ext.ooo.jurt.lib}"/>
+    <fileset dir="${openoffice.root.dir}/basis-link/ure-link/share/java">
+      <include name="**/*.jar" />
+      <include name="**/*.zip" />
+    </fileset>
+    <fileset dir="${openoffice.root.dir}/basis-link/program/classes">
+      <include name="**/*.jar" />
+      <include name="**/*.zip" />
+    </fileset>
+
 	<!-- -->
     <pathelement location="${dist.dir}/resource"/>
     <pathelement location="${dist.dir}/rules"/>
@@ -79,7 +84,7 @@
   <!-- TEST -->
   <!-- ================================================================== -->
 
-  <target name="test" depends="clean,init,build,ooo-build" description="Runs unit tests">
+  <target name="test" description="Runs unit tests">
   	
 	<mkdir dir="${junit.classes.dir}"/>
 	<mkdir dir="${dist.dir}/report/tests"/>
@@ -241,16 +246,6 @@
 	  debug="${debug}">
       <classpath refid="classpath"/>
     </javac>
-  	<javac encoding="${encoding}"
-      srcdir="${src.test.dir}"
-      includes="de/danielnaber/languagetool/rules/patterns/PatternRuleTest.java
-  		de/danielnaber/languagetool/ValidateXMLTest.java
-  		de/danielnaber/languagetool/XMLValidator.java"
-      destdir="${build.classes}"
-      target="${target}"
-	  debug="${debug}">
-      <classpath refid="classpath"/>
-    </javac>
 
   </target>
 
@@ -264,14 +259,21 @@
 	  		<include name="*.properties" />
 	  	</fileset>
 	</copy>
-
+	
+	<copy todir="${build.classes}/resource">
+	  	<fileset dir="src/resource"/>
+	</copy>
+	
+	<copy todir="${build.classes}/rules">
+	  	<fileset dir="src/rules"/>
+	</copy>
 	<jar
 		jarfile="${dist.dir}/${final.name}.jar"
 		basedir="${build.classes}"
 		excludes="**/*.java">
 		<manifest>
 			<attribute name="Main-Class" value="de.danielnaber.languagetool.Main" />
-			<attribute name="Class-Path" value="./ ./rules ./resource ./${stempelator.lib} ./jWordSplitter.jar" />
+			<attribute name="Class-Path" value="../ ../rules ../resource ./${stempelator.lib} ./jWordSplitter.jar" />
 			<attribute name="Implementation-Version" value="${version}" />
 			<attribute name="Implementation-Date" value="${date}" />
 		</manifest>
@@ -283,7 +285,7 @@
 		includes="de/danielnaber/languagetool/gui/*.class">
 		<manifest>
 			<attribute name="Main-Class" value="de.danielnaber.languagetool.gui.Main" />
-			<attribute name="Class-Path" value="./LanguageTool.jar ./ ./resource ./rules ./${stempelator.lib}  ./jdic.jar ./jaminid.jar ./jWordSplitter.jar" />
+			<attribute name="Class-Path" value="./LanguageTool.jar ../ ../resource ../rules ./${stempelator.lib}  ./jdic.jar ./jaminid.jar ./jWordSplitter.jar" />
 			<attribute name="Implementation-Version" value="${version}" />
 			<attribute name="Implementation-Date" value="${date}" />
 		</manifest>
@@ -350,12 +352,12 @@
   	<jar
 		jarfile="${dist.dir}/LanguageTool.uno.jar"
 		basedir="${build.classes}"
-		includes="de/danielnaber/languagetool/**/*.class,de/danielnaber/languagetool/**/*.properties">
+		includes="de/danielnaber/languagetool/openoffice/**/*.class">
   		<manifest>
 	  		<attribute name="RegistrationClassName" value="de.danielnaber.languagetool.openoffice.Main" />
 			<attribute name="Implementation-Version" value="${version}" />
 			<attribute name="Implementation-Date" value="${date}" />
-			<attribute name="Class-Path" value=". ./rules ./resource ./${stempelator.lib} ./jWordSplitter.jar" />
+			<attribute name="Class-Path" value="../ /usr/share/languagetool/lib/LanguageTool.jar /usr/share/morfologik-stemming/lib/morfologik-stemming-nodict.jar /usr/share/jwordsplitter/lib/jWordSplitter.jar" />
 			<attribute name="Uno-Type-Path" value="" />
   		</manifest>
 	</jar>
@@ -393,14 +395,14 @@
   	<zip compress="yes" destfile="${dist.dir}/LanguageTool-${version}.oxt"
   		basedir="${dist.dir}"
   		includes="**"
-  		excludes="deploy.sh,undeploy.sh,turn.sh,jdic.jar,printlocales.jar,linux/**,windows/**,sunos/**,mac/**" />
+  		excludes="deploy.sh,undeploy.sh,turn.sh,jdic.jar,printlocales.jar,LanguageToolGUI.jar,LanguageTool.jar,linux/**,windows/**,sunos/**,mac/**" />
   </target>
 
   <target name="fast" depends="ooo-internal" description="like dist, but without ZIP compression (for development only)">
   	<zip compress="no" destfile="${dist.dir}/LanguageTool-${version}.oxt"
   		basedir="${dist.dir}"
   		includes="**" 
-  		excludes="deploy.sh,undeploy.sh,turn.sh,jdic.jar,printlocales.jar,linux/**,windows/**,sunos/**,mac/**" />
+  		excludes="deploy.sh,undeploy.sh,turn.sh,jdic.jar,printlocales.jar,LanguageToolGUI.jar,LanguageTool.jar,linux/**,windows/**,sunos/**,mac/**" />
   </target>
 
   <target name="ooo-internal" depends="clean,init,build,ooo-build">
@@ -437,103 +439,6 @@
     <copy todir="${dist.dir}/">
         <fileset dir="openoffice/" includes="images/**" />
     </copy>
-  	<copy todir="${dist.dir}/rules">
-	  	<fileset dir="src/rules" excludes="CVS" />
-	</copy>
-		
-	<copy file="./README.txt" todir="${dist.dir}" />
-	<copy file="./CHANGES.txt" todir="${dist.dir}" />
-	<copy file="./COPYING.txt" todir="${dist.dir}" />
-  	<replace file="${dist.dir}/README.txt" token="###VERSION###" value="${version}"></replace>
-  	<replace file="${dist.dir}/README.txt" token="###DATE###" value="${date}"></replace>
-  	<copy file="${ext.dir}/LGPL.txt" todir="${dist.dir}" />
-  	<copy file="${ext.dir}/Morfologik-nodict-license.txt" tofile="${dist.dir}/Morfologik-license.txt" />
-  	<copy file="${ext.dir}/CPL.txt" tofile="${dist.dir}/junit-license.txt" />
-  	<copy file="${ext.dir}/Apache-License.txt" tofile="${dist.dir}/jWordSplitter-license.txt" />
-  	
-	<filter filtersFile="build.properties"/>
-  	<copy file="${ext.dir}/README" tofile="${dist.dir}/README-license.txt" 
-  	filtering="true"/>
-  		
-	<copy file="${ext.stempelator.lib}" todir="${dist.dir}" />
-	<copy file="${ext.jdic.lib}" todir="${dist.dir}" />
-	<copy file="${ext.jaminid.lib}" todir="${dist.dir}" />
-	<copy file="${ext.jwordsplitter.lib}" todir="${dist.dir}" />
-	<copy file="${ext.junit.lib}" todir="${dist.dir}" />
-
- 	<copy todir="${dist.dir}/linux">
- 		  	<fileset dir="${ext.dir}/linux"/>
-	</copy>
- 	<copy todir="${dist.dir}/windows">
-   		  	<fileset dir="${ext.dir}/windows"/>
- 	</copy>
- 	<copy todir="${dist.dir}/sunos">
-   		  	<fileset dir="${ext.dir}/sunos"/>
- 	</copy>
- 	<copy todir="${dist.dir}/mac">
-   		  	<fileset dir="${ext.dir}/mac"/>
- 	</copy>
-  	<zip compress="yes" destfile="${dist.dir}/standalone-libs.zip"
-  	  basedir="${dist.dir}"
-  	  includes="jdic.jar,linux/**,windows/**,sunos/**,mac/**" />
-
- 	<copy file="${src.scripts.dir}/testrules.sh" todir="${dist.dir}" />
- 	<copy file="${src.scripts.dir}/testrules.bat" todir="${dist.dir}" />
-
- 	<copy file="${resource.dir}/api-output.dtd" todir="${dist.dir}/resource" />
-  	<copy file="${resource.dir}/disambiguation.dtd" todir="${dist.dir}/resource" />
- 	<copy file="${resource.dir}/en/tagset.txt" todir="${dist.dir}/resource/en" />
-	<copy file="${resource.dir}/en/english.dict" todir="${dist.dir}/resource/en" />
-	<copy file="${resource.dir}/en/english.info" todir="${dist.dir}/resource/en" />
-	<copy file="${resource.dir}/en/english_synth.dict" todir="${dist.dir}/resource/en" />  	
-	<copy file="${resource.dir}/en/english_synth.info" todir="${dist.dir}/resource/en" />  	
-  	<copy file="${resource.dir}/en/english_tags.txt" todir="${dist.dir}/resource/en" />
-  	<copy file="${resource.dir}/en/disambiguation.xml" todir="${dist.dir}/resource/en" />
-  	<copy file="${resource.dir}/en/compounds.txt" todir="${dist.dir}/resource/en" />
-  	<copy file="${resource.dir}/it/italian.dict" todir="${dist.dir}/resource/it" />
-  	<copy file="${resource.dir}/it/italian.info" todir="${dist.dir}/resource/it" />
-  	<copy file="${resource.dir}/it/readme-morph-it.txt" todir="${dist.dir}/resource/it" />
- 	<copy file="${resource.dir}/fr/french.dict" todir="${dist.dir}/resource/fr" />
- 	<copy file="${resource.dir}/fr/french.info" todir="${dist.dir}/resource/fr" />
-  	<copy file="${resource.dir}/fr/readme.txt" todir="${dist.dir}/resource/fr" />
- 	<copy file="${resource.dir}/fr/disambiguation.xml" todir="${dist.dir}/resource/fr" />
-	<copy file="${resource.dir}/es/spanish.dict" todir="${dist.dir}/resource/es" />
-	<copy file="${resource.dir}/es/spanish.info" todir="${dist.dir}/resource/es" />
-  	<copy file="${resource.dir}/es/etiquetas-eagles.txt" todir="${dist.dir}/resource/es" />  	
-	<copy file="${resource.dir}/de/german.dict" todir="${dist.dir}/resource/de" />
-	<copy file="${resource.dir}/de/german.info" todir="${dist.dir}/resource/de" />
-  	<copy file="${resource.dir}/de/compounds.txt" todir="${dist.dir}/resource/de" />
-	<copy file="${resource.dir}/de/added.txt" todir="${dist.dir}/resource/de" />
-	<copy file="${resource.dir}/pl/polish.dict" todir="${dist.dir}/resource/pl" />
-	<copy file="${resource.dir}/pl/polish.info" todir="${dist.dir}/resource/pl" />
-  	<copy file="${resource.dir}/pl/polish_synth.dict" todir="${dist.dir}/resource/pl" />
-  	<copy file="${resource.dir}/pl/polish_synth.info" todir="${dist.dir}/resource/pl" />
-  	<copy file="${resource.dir}/pl/polish_tags.txt" todir="${dist.dir}/resource/pl" />
-  	<copy file="${resource.dir}/pl/compounds.txt" todir="${dist.dir}/resource/pl" />
-  	<copy file="${resource.dir}/pl/disambiguation.xml" todir="${dist.dir}/resource/pl" />
-	<copy file="${resource.dir}/nl/dutch.dict" todir="${dist.dir}/resource/nl" />
-	<copy file="${resource.dir}/nl/dutch.info" todir="${dist.dir}/resource/nl" />
-	<copy file="${resource.dir}/nl/dutch_synth.dict" todir="${dist.dir}/resource/nl" />
-	<copy file="${resource.dir}/nl/dutch_synth.info" todir="${dist.dir}/resource/nl" />
-	<copy file="${resource.dir}/nl/dutch_tags.txt" todir="${dist.dir}/resource/nl" />
-  	<copy file="${resource.dir}/nl/disambiguation.xml" todir="${dist.dir}/resource/nl" />
- 	<copy file="${resource.dir}/sv/swedish.info" todir="${dist.dir}/resource/sv" />
-	<copy file="${resource.dir}/sv/swedish.dict" todir="${dist.dir}/resource/sv" />
-  	<copy file="${resource.dir}/sv/compounds.txt" todir="${dist.dir}/resource/sv" />
-  	<copy file="${resource.dir}/uk/ukrainian.dict" todir="${dist.dir}/resource/uk" />
-	<copy file="${resource.dir}/ru/russian.dict" todir="${dist.dir}/resource/ru" />
-	<copy file="${resource.dir}/ru/russian.info" todir="${dist.dir}/resource/ru" />
-	<copy file="${resource.dir}/ru/russian_tags.txt" todir="${dist.dir}/resource/ru" />
-	<copy file="${resource.dir}/ru/compounds_ru.txt" todir="${dist.dir}/resource/ru" />
-	<copy file="${resource.dir}/ro/romanian.dict" todir="${dist.dir}/resource/ro" />
-  	<copy file="${resource.dir}/ro/romanian.info" todir="${dist.dir}/resource/ro" />
-	<copy file="${resource.dir}/ro/romanian_synth.dict" todir="${dist.dir}/resource/ro" />  	
-	<copy file="${resource.dir}/ro/romanian_synth.info" todir="${dist.dir}/resource/ro" />
-  	<copy file="${resource.dir}/ro/romanian_tags.txt" todir="${dist.dir}/resource/ro" />
-  	<copy file="${resource.dir}/ro/disambiguation.xml" todir="${dist.dir}/resource/ro" />
-  	<copy file="${resource.dir}/sk/slovak.dict" todir="${dist.dir}/resource/sk" />  	
-	<copy file="${resource.dir}/sk/compounds.txt" todir="${dist.dir}/resource/sk" />
-  	<copy file="${resource.dir}/xx/dummy.dict" tofile="${dist.dir}/resource/cs/czech.dict" />
   </target>
 
   <target name="dict-de" description="Build the German tagger dictionary from Morphy data">
