--- Makefile.am.orig	2005-11-12 14:02:19.000000000 -0600
+++ Makefile.am	2005-11-12 14:02:42.000000000 -0600
@@ -74,11 +74,11 @@
 libcairojava_la_SOURCES = $(java_cairo_source_files)
 libcairojava_la_GCJFLAGS = -fjni -I$(srcdir)/src/java -Isrc/java --classpath=$(JGJAR):$(srcdir)/src/java
 libcairojava_la_LDFLAGS = $(JGJAVA_LIBS) -release $(apiversion)
-endif
-
+else
 libcairojni_la_SOURCES = $(jni_cairo_source_files)
-libcairojni_la_CFLAGS = $(CAIRO_CFLAGS) $(FONTCONFIG_CFLAGS) $(GLIB_CFLAGS) $(JGJAVA_CFLAGS) -I$(includedir)
+libcairojni_la_CFLAGS = $(CAIRO_CFLAGS) $(FONTCONFIG_CFLAGS) $(GLIB_CFLAGS) $(JGJAVA_CFLAGS) -I$(includedir) $(JNI_INCLUDES)
 libcairojni_la_LDFLAGS = $(CAIRO_LIBS) $(FONTCONFIG_LIBS) $(JGJNI_LIBS) -release $(apiversion)
+endif
 
 # automake should support generating versioned jars
 cairo_jardir = $(jardir)
