Index: plexus-compiler-1.3.3_pre20060602/plexus-compilers/plexus-compiler-javac/src/main/java/org/codehaus/plexus/compiler/javac/Commandline.java
===================================================================
--- plexus-compiler-1.3.3_pre20060602/plexus-compilers/plexus-compiler-javac/src/main/java/org/codehaus/plexus/compiler/javac/Commandline.java	(révision 6983)
+++ plexus-compiler-1.3.3_pre20060602/plexus-compilers/plexus-compiler-javac/src/main/java/org/codehaus/plexus/compiler/javac/Commandline.java	(copie de travail)
@@ -22,7 +22,7 @@
 
 import org.codehaus.plexus.compiler.javac.shell.CmdShell;
 import org.codehaus.plexus.compiler.javac.shell.CommandShell;
-import org.codehaus.plexus.compiler.javac.shell.Shell;
+import org.codehaus.plexus.util.cli.shell.Shell;
 
 /**
  * <p>
@@ -38,7 +38,7 @@
     extends org.codehaus.plexus.util.cli.Commandline
 {
 
-    private Shell shell;
+    private org.codehaus.plexus.util.cli.shell.Shell   shell;
 
     public Commandline()
     {
@@ -56,6 +56,7 @@
 
         //If this is windows set the shell to command.com or cmd.exe with correct arguments.
         if ( os.indexOf( WINDOWS ) != -1 )
+
         {
             if ( os.indexOf( "95" ) != -1 || os.indexOf( "98" ) != -1 || os.indexOf( "Me" ) != -1 )
             {
@@ -73,7 +74,7 @@
      */
     public String[] getShellCommandline()
     {
-
+        
         if ( getShell() == null )
         {
             if ( executable != null )
@@ -91,7 +92,18 @@
         }
         else
         {
-            return (String[]) getShell().getShellCommandLine( executable, getArguments() ).toArray( new String[0] );
+            /* */
+            String args[]= getArguments();
+            int tab = 0 ; 
+            String mytab [] = new String [1+args.length]; 
+            mytab[tab]=executable;
+            int i = 0 ;
+            for ( i =0;i<args.length;i++){
+                tab += 1;
+                mytab[tab] = args[i];
+            }
+            /* */ 
+            return (String[]) getShell().getShellCommandLine( args).toArray( new String[0] );
         }
     }
 
@@ -101,7 +113,7 @@
      * @param shell
      * @since 1.2
      */
-    public void setShell( Shell shell )
+    public void setShell( org.codehaus.plexus.util.cli.shell.Shell  shell )
     {
         this.shell = shell;
     }
@@ -113,7 +125,7 @@
      */
     public Shell getShell()
     {
-        return shell;
+        return  shell;
     }
 
 }
Index: plexus-compiler-1.3.3_pre20060602/plexus-compilers/plexus-compiler-javac/src/main/java/org/codehaus/plexus/compiler/javac/shell/Shell.java
===================================================================
--- plexus-compiler-1.3.3_pre20060602/plexus-compilers/plexus-compiler-javac/src/main/java/org/codehaus/plexus/compiler/javac/shell/Shell.java	(révision 6983)
+++ plexus-compiler-1.3.3_pre20060602/plexus-compilers/plexus-compiler-javac/src/main/java/org/codehaus/plexus/compiler/javac/shell/Shell.java	(copie de travail)
@@ -25,6 +25,7 @@
 
 /**
  * <p>
+ *
  * Class with patches copied from plexus-utils with fix for PLX-161,
  * as we can not upgrade plexus-utils until it's upgraded in core Maven
  * </p>
@@ -33,7 +34,7 @@
  *
  * @author <a href="mailto:carlos@apache.org">Carlos Sanchez</a>
  */
-public class Shell
+public class Shell extends org.codehaus.plexus.util.cli.shell.Shell
 {
     private String shellCommand;
 
