diff -r ce3ea0b5a8c9 acinclude.m4
--- a/acinclude.m4	Thu Oct 16 01:28:18 2008 +0100
+++ b/acinclude.m4	Thu Oct 16 01:32:45 2008 +0100
@@ -847,13 +847,13 @@
   AM_CONDITIONAL(SHARK_BUILD, test "x${use_shark}" = xyes)
 ])
 
-AC_DEFUN([AC_CHECK_WITH_CACAO],
+AC_DEFUN([AC_CHECK_ENABLE_CACAO],
 [
   AC_MSG_CHECKING(whether to use CACAO as VM)
-  AC_ARG_WITH([cacao],
-	      [AS_HELP_STRING(--with-cacao,use CACAO as VM [[default=no]])],
+  AC_ARG_ENABLE([cacao],
+	      [AS_HELP_STRING(--enable-cacao,use CACAO as VM [[default=no]])],
   [
-    WITH_CACAO=yes
+    WITH_CACAO="${enableval}"
   ],
   [
     WITH_CACAO=no
diff -r ce3ea0b5a8c9 configure.ac
--- a/configure.ac	Thu Oct 16 01:28:18 2008 +0100
+++ b/configure.ac	Thu Oct 16 01:32:45 2008 +0100
@@ -249,7 +249,7 @@
 WITH_NETBEANS_PROFILER_SRC_ZIP
 WITH_NETBEANS_BASIC_CLUSTER_SRC_ZIP
 WITH_ALT_JAR_BINARY
-AC_CHECK_WITH_CACAO
+AC_CHECK_ENABLE_CACAO
 AC_CHECK_WITH_CACAO_HOME
 AC_CHECK_WITH_CACAO_SRC_ZIP
 ENABLE_OPTIMIZATIONS
