diff -r a5c32475a2e8 Makefile.am
--- a/Makefile.am	Fri Apr 04 04:40:21 2008 -0400
+++ b/Makefile.am	Fri May 30 01:10:44 2008 +0100
@@ -104,6 +104,11 @@ ICEDTEA_ENV += \
 	"ALT_HOTSPOT_IMPORT_PATH=$(CACAO)"
 endif
 
+if !ENABLE_DOCS
+ICEDTEA_ENV += \
+	"NO_DOCS=true"
+endif
+
 env:
 	@echo 'unset JAVA_HOME'
 	@echo 'export $(ICEDTEA_ENV)'
@@ -152,6 +157,11 @@ if WITH_CACAO
 if WITH_CACAO
 ICEDTEA_ENV_ECJ += \
 	"ALT_HOTSPOT_IMPORT_PATH=$(CACAO)"
+endif
+
+if !ENABLE_DOCS
+ICEDTEA_ENV_ECJ += \
+	"NO_DOCS=true"
 endif
 
 env-ecj:
diff -r a5c32475a2e8 Makefile.in
--- a/Makefile.in	Fri Apr 04 04:40:21 2008 -0400
+++ b/Makefile.in	Fri May 30 01:10:44 2008 +0100
@@ -34,10 +34,16 @@ host_triplet = @host@
 @WITH_CACAO_TRUE@am__append_1 = \
 @WITH_CACAO_TRUE@	"ALT_HOTSPOT_IMPORT_PATH=$(CACAO)"
 
-@WITH_CACAO_TRUE@am__append_2 = \
+@ENABLE_DOCS_FALSE@am__append_2 = \
+@ENABLE_DOCS_FALSE@	"NO_DOCS=true"
+
+@WITH_CACAO_TRUE@am__append_3 = \
 @WITH_CACAO_TRUE@	"ALT_HOTSPOT_IMPORT_PATH=$(CACAO)"
 
-@WITH_CACAO_TRUE@am__append_3 = \
+@ENABLE_DOCS_FALSE@am__append_4 = \
+@ENABLE_DOCS_FALSE@	"NO_DOCS=true"
+
+@WITH_CACAO_TRUE@am__append_5 = \
 @WITH_CACAO_TRUE@	patches/icedtea-cacao.patch
 
 subdir = .
@@ -289,7 +295,7 @@ ICEDTEA_ENV = "ALT_JDK_IMPORT_PATH=$(ICE
 	"ZERO_ARCHFLAG=$(ZERO_ARCHFLAG)" \
 	"LIBFFI_CFLAGS=$(LIBFFI_CFLAGS)" "LIBFFI_LIBS=$(LIBFFI_LIBS)" \
 	"FREETYPE2_HEADERS=$(FREETYPE2_CFLAGS)" \
-	"FT2_LIB=$(FREETYPE2_LIBS)" $(am__append_1)
+	"FT2_LIB=$(FREETYPE2_LIBS)" $(am__append_1) $(am__append_2)
 
 # OpenJDK ecj build environment.
 ICEDTEA_BUILD_DIR_ECJ = \
@@ -326,7 +332,7 @@ ICEDTEA_ENV_ECJ = "ALT_JDK_IMPORT_PATH=$
 	"ZERO_ARCHFLAG=$(ZERO_ARCHFLAG)" \
 	"LIBFFI_CFLAGS=$(LIBFFI_CFLAGS)" "LIBFFI_LIBS=$(LIBFFI_LIBS)" \
 	"FREETYPE2_HEADERS=$(FREETYPE2_CFLAGS)" \
-	"FT2_LIB=$(FREETYPE2_LIBS)" $(am__append_2)
+	"FT2_LIB=$(FREETYPE2_LIBS)" $(am__append_3) $(am__append_4)
 
 # OpenJDK Source Preparation Targets
 # ==================================
@@ -387,7 +393,7 @@ ICEDTEA_PATCHES = $(ZERO_PATCHES_COND) \
 	patches/icedtea-print-lsb-release.patch \
 	patches/icedtea-jpegclasses.patch \
 	patches/icedtea-arm-uname.patch $(GCC_PATCH) \
-	$(FAST_BUILD_PATCH) $(DISTRIBUTION_PATCHES) $(am__append_3)
+	$(FAST_BUILD_PATCH) $(DISTRIBUTION_PATCHES) $(am__append_5)
 
 # Patch OpenJDK for plug replacements and ecj.
 ICEDTEA_ECJ_PATCH = patches/icedtea-ecj.patch
diff -r a5c32475a2e8 configure
--- a/configure	Fri Apr 04 04:40:21 2008 -0400
+++ b/configure	Fri May 30 01:10:44 2008 +0100
@@ -722,6 +722,8 @@ SYSTEM_ICEDTEA_DIR
 SYSTEM_ICEDTEA_DIR
 SYSTEM_ANT_DIR
 ENABLE_PLUGIN
+ENABLE_DOCS_TRUE
+ENABLE_DOCS_FALSE
 WITH_ICEDTEA_TRUE
 WITH_ICEDTEA_FALSE
 WITH_OPENJDK_TRUE
@@ -1417,6 +1419,7 @@ Optional Features:
   --disable-dependency-tracking  speeds up one-time build
   --enable-dependency-tracking   do not reject slow dependency extractors
   --disable-gcjwebplugin  Disable compilation of browser plugin
+  --disable-docs          Disable generation of documentation
   --enable-fast-build     optimize for quick building: use -O0 and do not
                           build documentation
   --enable-zero           use zero-assembler port on non-zero platforms
@@ -5539,6 +5542,22 @@ fi
 
 
 
+# Check whether --enable-docs was given.
+if test "${enable_docs+set}" = set; then
+  enableval=$enable_docs; ENABLE_DOCS="$val"
+else
+  ENABLE_DOCS='yes'
+fi
+
+ if test x$ENABLE_DOCS = xyes; then
+  ENABLE_DOCS_TRUE=
+  ENABLE_DOCS_FALSE='#'
+else
+  ENABLE_DOCS_TRUE='#'
+  ENABLE_DOCS_FALSE=
+fi
+
+
 
 # Check whether --with-icedtea was given.
 if test "${with_icedtea+set}" = set; then
@@ -11774,6 +11793,13 @@ if test -z "${am__fastdepCXX_TRUE}" && t
   { { echo "$as_me:$LINENO: error: conditional \"am__fastdepCXX\" was never defined.
 Usually this means the macro was only invoked conditionally." >&5
 echo "$as_me: error: conditional \"am__fastdepCXX\" was never defined.
+Usually this means the macro was only invoked conditionally." >&2;}
+   { (exit 1); exit 1; }; }
+fi
+if test -z "${ENABLE_DOCS_TRUE}" && test -z "${ENABLE_DOCS_FALSE}"; then
+  { { echo "$as_me:$LINENO: error: conditional \"ENABLE_DOCS\" was never defined.
+Usually this means the macro was only invoked conditionally." >&5
+echo "$as_me: error: conditional \"ENABLE_DOCS\" was never defined.
 Usually this means the macro was only invoked conditionally." >&2;}
    { (exit 1); exit 1; }; }
 fi
@@ -12557,6 +12583,8 @@ SYSTEM_ICEDTEA_DIR!$SYSTEM_ICEDTEA_DIR$a
 SYSTEM_ICEDTEA_DIR!$SYSTEM_ICEDTEA_DIR$ac_delim
 SYSTEM_ANT_DIR!$SYSTEM_ANT_DIR$ac_delim
 ENABLE_PLUGIN!$ENABLE_PLUGIN$ac_delim
+ENABLE_DOCS_TRUE!$ENABLE_DOCS_TRUE$ac_delim
+ENABLE_DOCS_FALSE!$ENABLE_DOCS_FALSE$ac_delim
 WITH_ICEDTEA_TRUE!$WITH_ICEDTEA_TRUE$ac_delim
 WITH_ICEDTEA_FALSE!$WITH_ICEDTEA_FALSE$ac_delim
 WITH_OPENJDK_TRUE!$WITH_OPENJDK_TRUE$ac_delim
@@ -12632,7 +12660,7 @@ LTLIBOBJS!$LTLIBOBJS$ac_delim
 LTLIBOBJS!$LTLIBOBJS$ac_delim
 _ACEOF
 
-  if test `sed -n "s/.*$ac_delim\$/X/p" conf$$subs.sed | grep -c X` = 83; then
+  if test `sed -n "s/.*$ac_delim\$/X/p" conf$$subs.sed | grep -c X` = 85; then
     break
   elif $ac_last_try; then
     { { echo "$as_me:$LINENO: error: could not make $CONFIG_STATUS" >&5
diff -r a5c32475a2e8 configure.ac
--- a/configure.ac	Fri Apr 04 04:40:21 2008 -0400
+++ b/configure.ac	Fri May 30 01:10:44 2008 +0100
@@ -100,6 +100,12 @@ AC_ARG_ENABLE([gcjwebplugin],
 	      		      [Disable compilation of browser plugin])],
 	      [ENABLE_PLUGIN="$val"], [ENABLE_PLUGIN='yes'])
 AC_SUBST(ENABLE_PLUGIN)
+
+AC_ARG_ENABLE([docs],
+	      [AS_HELP_STRING([--disable-docs], 
+	      		      [Disable generation of documentation])],
+	      [ENABLE_DOCS="${withval}"], [ENABLE_DOCS='yes'])
+AM_CONDITIONAL([ENABLE_DOCS], [test x$ENABLE_DOCS = xyes])
 
 AC_ARG_WITH([icedtea],
         [AS_HELP_STRING([--with-icedtea],
