diff -r 6cc69d786ebb configure.ac
--- a/configure.ac	Sun Oct 19 18:06:54 2008 +0200
+++ b/configure.ac	Mon Oct 20 10:51:00 2008 +0100
@@ -125,13 +125,13 @@
 AC_ARG_ENABLE([gcjwebplugin],
               [AS_HELP_STRING([--enable-gcjwebplugin],
                               [Build the GCJ Web Plugin plugin (deprecated)])],
-              [enable_gcjwebplugin="yes"], [enable_gcjwebplugin="no"])
+              [enable_gcjwebplugin="${enableval}"], [enable_gcjwebplugin="no"])
 AM_CONDITIONAL(ENABLE_GCJWEBPLUGIN, test "x${enable_gcjwebplugin}" = "xyes")
 
 AC_ARG_ENABLE([pulse-java],
               [AS_HELP_STRING([--enable-pulse-java],
                               [Enable pulse-java - an audio mixer spi that uses PulseAudio])],
-              [enable_pulse_java="yes"], [enable_pulse_java="no"])
+              [enable_pulse_java="${enableval}"], [enable_pulse_java="no"])
 AM_CONDITIONAL(ENABLE_PULSE_JAVA, test "x${enable_pulse_java}" = "xyes")
 
 AC_ARG_ENABLE([docs],
@@ -143,7 +143,7 @@
 AC_ARG_ENABLE([visualvm],
               [AS_HELP_STRING([--enable-visualvm],
 			                  [Enable compilation of visualvm.])],
-              [enable_visualvm="yes"], [enable_visualvm="no"])
+              [enable_visualvm="${enableval}"], [enable_visualvm="no"])
 AM_CONDITIONAL(WITH_VISUALVM, test "x${enable_visualvm}" = "xyes")
 
 if test "x${enable_visualvm}" = "xyes"
