--- /home/andrew/projects/openjdk/icedtea6/configure.ac	2008-05-30 00:34:34.000000000 +0100
+++ configure.ac	2008-05-30 00:23:20.000000000 +0100
@@ -101,13 +101,13 @@
 AC_ARG_ENABLE([gcjwebplugin],
 	      [AS_HELP_STRING([--disable-gcjwebplugin], 
 	      		      [Disable compilation of browser plugin])],
-	      [ENABLE_PLUGIN="$val"], [ENABLE_PLUGIN='yes'])
+	      [ENABLE_PLUGIN="${enableval}"], [ENABLE_PLUGIN='yes'])
 AC_SUBST(ENABLE_PLUGIN)
 
 AC_ARG_ENABLE([docs],
 	      [AS_HELP_STRING([--disable-docs], 
 	      		      [Disable generation of documentation])],
-	      [ENABLE_DOCS="$val"], [ENABLE_DOCS='yes'])
+	      [ENABLE_DOCS="${enableval}"], [ENABLE_DOCS='yes'])
 AM_CONDITIONAL([ENABLE_DOCS], [test x$ENABLE_DOCS = xyes])
 
 AC_ARG_WITH([icedtea],
