diff -r ecfb2532387c Makefile.am
--- a/Makefile.am	Mon Oct 27 15:56:05 2008 -0400
+++ b/Makefile.am	Mon Oct 27 21:16:43 2008 +0000
@@ -140,7 +140,6 @@
         "BUILD_NUMBER=$(OPENJDK_VERSION)" \
 	"JDK_UPDATE_VERSION=$(JDK_UPDATE_VERSION)" \
         "JRE_RELEASE_VERSION=1.6.0_$(COMBINED_VERSION)" \
-        "HOTSPOT_RELEASE_VERSION=1.6.0_$(COMBINED_VERSION)" \
 	"LANG=C" \
 	"PATH=$(abs_top_builddir)/bootstrap/jdk1.6.0/bin:/usr/bin:/bin:/usr/sbin:/sbin" \
 	"ALT_BOOTDIR=$(ICEDTEA_BOOT_DIR)" \
@@ -210,7 +209,6 @@
 	"BUILD_NUMBER=$(OPENJDK_VERSION)" \
 	"JDK_UPDATE_VERSION=$(JDK_UPDATE_VERSION)" \
         "JRE_RELEASE_VERSION=1.6.0_$(COMBINED_VERSION)" \
-        "HOTSPOT_RELEASE_VERSION=1.6.0_$(COMBINED_VERSION)" \
 	"LANG=C" \
 	"PATH=$(abs_top_builddir)/bootstrap/jdk1.6.0/bin:/usr/bin:/bin:/usr/sbin:/sbin" \
 	"ALT_BOOTDIR=$(ICEDTEA_BOOT_DIR)" \
