diff -urNp icedtea6-1.3.org/configure.ac icedtea6-1.3/configure.ac
--- icedtea6-1.3.org/configure.ac	2008-10-15 18:33:06.000000000 +0200
+++ icedtea6-1.3/configure.ac	2008-10-19 13:21:28.000000000 +0200
@@ -125,13 +125,13 @@ AM_CONDITIONAL(ENABLE_PLUGIN, test "x${e
 AC_ARG_ENABLE([liveconnect],
               [AS_HELP_STRING([--enable-liveconnect],
                               [Enable experimental LiveConnect plugin])],
-              [enable_liveconnect="yes"], [enable_liveconnect="no"])
+              [enable_liveconnect="${enableval}"], [enable_liveconnect="no"])
 AM_CONDITIONAL(ENABLE_LIVECONNECT, test "x${enable_liveconnect}" = "xyes")
 
 AC_ARG_ENABLE([pulse-java],
               [AS_HELP_STRING([--enable-pulse-java],
                               [Enable pulse-java - an audio mixer spi that uses PulseAudio])],
-              [enable_pulse_java="yes"], [enable_pulse_java="no"])
+              [enable_pulse_java="${enableval}"], [enable_pulse_java="no"])
 AM_CONDITIONAL(ENABLE_PULSE_JAVA, test "x${enable_pulse_java}" = "xyes")
 
 AC_ARG_ENABLE([docs],
@@ -143,7 +143,7 @@ AM_CONDITIONAL([ENABLE_DOCS], [test x$EN
 AC_ARG_ENABLE([visualvm],
               [AS_HELP_STRING([--enable-visualvm],
 			                  [Enable compilation of visualvm.])],
-              [enable_visualvm="yes"], [enable_visualvm="no"])
+              [enable_visualvm="${enableval}"], [enable_visualvm="no"])
 AM_CONDITIONAL(WITH_VISUALVM, test "x${enable_visualvm}" = "xyes")
 
 if test "x${enable_visualvm}" = "xyes"
