--- javac.in	2008-06-30 22:06:01.000000000 +0100
+++ javac.in.new	2008-06-30 22:50:59.000000000 +0100
@@ -30,5 +30,7 @@
   NEW_ARGS=`echo $NEW_ARGS | sed -e 's/-Werror//1'`
 fi
 
+MEM_OPT=$(echo @JAVAC_MEM_OPT@|sed -e 's/-J//')
+
 CLASSPATH=@ECJ_JAR@${CLASSPATH:+:}$CLASSPATH \
-  @JAVA@ org.eclipse.jdt.internal.compiler.batch.Main -1.5 -nowarn $bcoption $NEW_ARGS
+  @JAVA@ ${MEM_OPT} org.eclipse.jdt.internal.compiler.batch.Main -1.5 -nowarn $bcoption $NEW_ARGS
