diff -Nru icedtea6-1.2.orig/configure.ac icedtea6-1.2/configure.ac
--- icedtea6-1.2.orig/configure.ac	2008-06-29 19:37:49.000000000 +0100
+++ icedtea6-1.2/configure.ac	2008-06-29 21:43:19.000000000 +0100
@@ -187,8 +187,8 @@
   AC_SUBST(RMIC)
   AM_CONDITIONAL(GCC_OLD, test x != x)
 else
-  FIND_JAVA
-  FIND_JAVAC
+  AC_PROG_JAVA
+  AC_PROG_JAVAC
   FIND_JAVAH
   FIND_JAR
   FIND_RMIC
@@ -201,6 +201,7 @@
   AC_CONFIG_FILES([javap], [chmod +x javap])
 fi
 fi
+CLASSPATH_JAVAC_MEM_CHECK
 WITH_OPENJDK_SRC_ZIP
 WITH_OPENJDK_SRC_DIR
 AC_CHECK_WITH_CACAO
diff -Nru icedtea6-1.2.orig/m4/ac_prog_javac.m4 icedtea6-1.2/m4/ac_prog_javac.m4
--- icedtea6-1.2.orig/m4/ac_prog_javac.m4	1970-01-01 01:00:00.000000000 +0100
+++ icedtea6-1.2/m4/ac_prog_javac.m4	2008-06-29 21:44:39.000000000 +0100
@@ -0,0 +1,49 @@
+dnl @synopsis AC_PROG_JAVAC
+dnl
+dnl AC_PROG_JAVAC tests an existing Java compiler. It uses the
+dnl environment variable JAVAC then tests in sequence various common
+dnl Java compilers. For political reasons, it starts with the free
+dnl ones.
+dnl
+dnl If you want to force a specific compiler:
+dnl
+dnl - at the configure.in level, set JAVAC=yourcompiler before calling
+dnl AC_PROG_JAVAC
+dnl
+dnl - at the configure level, setenv JAVAC
+dnl
+dnl You can use the JAVAC variable in your Makefile.in, with @JAVAC@.
+dnl
+dnl *Warning*: its success or failure can depend on a proper setting of
+dnl the CLASSPATH env. variable.
+dnl
+dnl TODO: allow to exclude compilers (rationale: most Java programs
+dnl cannot compile with some compilers like guavac).
+dnl
+dnl Note: This is part of the set of autoconf M4 macros for Java
+dnl programs. It is VERY IMPORTANT that you download the whole set,
+dnl some macros depend on other. Unfortunately, the autoconf archive
+dnl does not support the concept of set of macros, so I had to break it
+dnl for submission. The general documentation, as well as the sample
+dnl configure.in, is included in the AC_PROG_JAVA macro.
+dnl
+dnl @category Java
+dnl @author Stephane Bortzmeyer <bortzmeyer@pasteur.fr>
+dnl @version 2000-07-19
+dnl @license GPLWithACException
+dnl
+dnl Modified to remove jikes by Andrew John Hughes on 2008-02-11
+
+AC_DEFUN([AC_PROG_JAVAC],[
+AC_REQUIRE([AC_EXEEXT])dnl
+ECJ_OPTS="-warn:-deprecation,serial,unusedImport"
+JAVAC_OPTS="-Xlint:unchecked,cast,divzero,empty,finally,overrides"
+if test "x$JAVAPREFIX" = x; then
+        test "x$JAVAC" = x && AC_CHECK_PROGS(JAVAC, ["ecj$EXEEXT $ECJ_OPTS"] ["ecj-3.4$EXEEXT $ECJ_OPTS"] ["ecj-3.3$EXEEXT $ECJ_OPTS"] ["ecj-3.2$EXEEXT $ECJ_OPTS"] ["javac$EXEEXT $JAVAC_OPTS"] "gcj$EXEEXT -C")
+else
+        test "x$JAVAC" = x && AC_CHECK_PROGS(JAVAC, ["ecj$EXEEXT $ECJ_OPTS"] ["ecj-3.4$EXEEXT $ECJ_OPTS"] ["ecj-3.3$EXEEXT $ECJ_OPTS"] ["ecj-3.2$EXEEXT $ECJ_OPTS"] ["javac$EXEEXT $JAVAC_OPTS"] "gcj$EXEEXT -C", $JAVAPREFIX)
+fi
+test "x$JAVAC" = x && AC_MSG_ERROR([no acceptable Java compiler found in \$PATH])
+AC_PROG_JAVAC_WORKS
+AC_PROVIDE([$0])dnl
+])
diff -Nru icedtea6-1.2.orig/m4/ac_prog_javac_works.m4 icedtea6-1.2/m4/ac_prog_javac_works.m4
--- icedtea6-1.2.orig/m4/ac_prog_javac_works.m4	1970-01-01 01:00:00.000000000 +0100
+++ icedtea6-1.2/m4/ac_prog_javac_works.m4	2008-06-29 21:38:36.000000000 +0100
@@ -0,0 +1,44 @@
+dnl @synopsis AC_PROG_JAVAC_WORKS
+dnl
+dnl Internal use ONLY.
+dnl
+dnl Note: This is part of the set of autoconf M4 macros for Java
+dnl programs. It is VERY IMPORTANT that you download the whole set,
+dnl some macros depend on other. Unfortunately, the autoconf archive
+dnl does not support the concept of set of macros, so I had to break it
+dnl for submission. The general documentation, as well as the sample
+dnl configure.in, is included in the AC_PROG_JAVA macro.
+dnl
+dnl @category Java
+dnl @author Stephane Bortzmeyer <bortzmeyer@pasteur.fr>
+dnl @version 2000-07-19
+dnl @license GPLWithACException
+dnl
+dnl Modified to test for 1.5 by Andrew John Hughes on 2008-02-11
+
+AC_DEFUN([AC_PROG_JAVAC_WORKS],[
+AC_CACHE_CHECK([if $JAVAC works], ac_cv_prog_javac_works, [
+JAVA_TEST=Object.java
+CLASS_TEST=Object.class
+cat << \EOF > $JAVA_TEST
+/* [#]line __oline__ "configure" */
+package java.lang;
+
+public class Object
+{
+  static <T> void doStuff()
+  {
+  }
+}
+EOF
+if AC_TRY_COMMAND($JAVAC $JAVACFLAGS -source 1.5 -target 1.5 $JAVA_TEST) >/dev/null 2>&1; then
+  ac_cv_prog_javac_works=yes
+else
+  AC_MSG_ERROR([The Java compiler $JAVAC failed (see config.log, check the CLASSPATH?)])
+  echo "configure: failed program was:" >&AC_FD_CC
+  cat $JAVA_TEST >&AC_FD_CC
+fi
+rm -f $JAVA_TEST $CLASS_TEST
+])
+AC_PROVIDE([$0])dnl
+])
diff -Nru icedtea6-1.2.orig/m4/ac_prog_java.m4 icedtea6-1.2/m4/ac_prog_java.m4
--- icedtea6-1.2.orig/m4/ac_prog_java.m4	1970-01-01 01:00:00.000000000 +0100
+++ icedtea6-1.2/m4/ac_prog_java.m4	2008-06-29 21:38:54.000000000 +0100
@@ -0,0 +1,17 @@
+dnl Available from the GNU Autoconf Macro Archive at:
+dnl http://www.gnu.org/software/ac-archive/htmldoc/ac_prog_java.html
+dnl
+dnl Changes for gjdoc: check for "kaffe, gij, java", not only "kaffe, java"
+dnl Changes for GNU Classpath: check for "cacao, jamvm" as well
+dnl
+AC_DEFUN([AC_PROG_JAVA],[
+AC_REQUIRE([AC_EXEEXT])dnl
+if test x$JAVAPREFIX = x; then
+        test x$JAVA = x && AC_CHECK_PROGS(JAVA, cacao$EXEEXT jamvm$EXEEXT kaffe$EXEEXT gij$EXEEXT java$EXEEXT)
+else
+        test x$JAVA = x && AC_CHECK_PROGS(JAVA, cacao$EXEEXT jamvm$EXEEXT kaffe$EXEEXT gij$EXEEXT java$EXEEXT, $JAVAPREFIX)
+fi
+test x$JAVA = x && AC_MSG_ERROR([no acceptable Java virtual machine found in \$PATH])
+AC_PROG_JAVA_WORKS
+AC_PROVIDE([$0])dnl
+])
diff -Nru icedtea6-1.2.orig/m4/ac_prog_java_works.m4 icedtea6-1.2/m4/ac_prog_java_works.m4
--- icedtea6-1.2.orig/m4/ac_prog_java_works.m4	1970-01-01 01:00:00.000000000 +0100
+++ icedtea6-1.2/m4/ac_prog_java_works.m4	2008-06-29 21:38:58.000000000 +0100
@@ -0,0 +1,109 @@
+dnl Available from the GNU Autoconf Macro Archive at:
+dnl http://www.gnu.org/software/ac-archive/htmldoc/ac_prog_java_works.html
+dnl
+AC_DEFUN([AC_PROG_JAVA_WORKS], [
+AC_CHECK_PROG(uudecode, uudecode$EXEEXT, yes)
+if test x$uudecode = xyes; then
+AC_CACHE_CHECK([if uudecode can decode base 64 file], ac_cv_prog_uudecode_base64, [
+dnl /**
+dnl  * Test.java: used to test if java compiler works.
+dnl  */
+dnl public class Test
+dnl {
+dnl
+dnl public static void
+dnl main( String[] argv )
+dnl {
+dnl     System.exit (0);
+dnl }
+dnl
+dnl }
+cat << \EOF > Test.uue
+begin-base64 644 Test.class
+yv66vgADAC0AFQcAAgEABFRlc3QHAAQBABBqYXZhL2xhbmcvT2JqZWN0AQAE
+bWFpbgEAFihbTGphdmEvbGFuZy9TdHJpbmc7KVYBAARDb2RlAQAPTGluZU51
+bWJlclRhYmxlDAAKAAsBAARleGl0AQAEKEkpVgoADQAJBwAOAQAQamF2YS9s
+YW5nL1N5c3RlbQEABjxpbml0PgEAAygpVgwADwAQCgADABEBAApTb3VyY2VG
+aWxlAQAJVGVzdC5qYXZhACEAAQADAAAAAAACAAkABQAGAAEABwAAACEAAQAB
+AAAABQO4AAyxAAAAAQAIAAAACgACAAAACgAEAAsAAQAPABAAAQAHAAAAIQAB
+AAEAAAAFKrcAErEAAAABAAgAAAAKAAIAAAAEAAQABAABABMAAAACABQ=
+====
+EOF
+if uudecode$EXEEXT Test.uue; then
+        ac_cv_prog_uudecode_base64=yes
+else
+        echo "configure: __oline__: uudecode had trouble decoding base 64 file 'Test.uue'" >&AC_FD_CC
+        echo "configure: failed file was:" >&AC_FD_CC
+        cat Test.uue >&AC_FD_CC
+        ac_cv_prog_uudecode_base64=no
+fi
+rm -f Test.uue])
+fi
+if test x$ac_cv_prog_uudecode_base64 != xyes; then
+        rm -f Test.class
+        AC_MSG_WARN([I have to compile Test.class from scratch])
+        if test x$ac_cv_prog_javac_works = xno; then
+                AC_MSG_ERROR([Cannot compile java source. $JAVAC does not work properly])
+        fi
+        if test x$ac_cv_prog_javac_works = x; then
+                AC_PROG_JAVAC
+        fi
+fi
+AC_CACHE_CHECK(if $JAVA works, ac_cv_prog_java_works, [
+JAVA_TEST=Test.java
+CLASS_TEST=Test.class
+TEST=Test
+changequote(, )dnl
+cat << \EOF > $JAVA_TEST
+/* [#]line __oline__ "configure" */
+public class Test {
+public static void main (String args[]) {
+        System.exit (0);
+} }
+EOF
+changequote([, ])dnl
+if test x$ac_cv_prog_uudecode_base64 != xyes; then
+        if AC_TRY_COMMAND($JAVAC $JAVACFLAGS $JAVA_TEST) && test -s $CLASS_TEST; then
+                :
+        else
+          echo "configure: failed program was:" >&AC_FD_CC
+          cat $JAVA_TEST >&AC_FD_CC
+          AC_MSG_ERROR(The Java compiler $JAVAC failed (see config.log, check the CLASSPATH?))
+        fi
+fi
+if AC_TRY_COMMAND(CLASSPATH=.:$CLASSPATH $JAVA $JAVAFLAGS $TEST) >/dev/null 2>&1; then
+  ac_cv_prog_java_works=yes
+else
+  echo "configure: failed program was:" >&AC_FD_CC
+  cat $JAVA_TEST >&AC_FD_CC
+  AC_MSG_ERROR(The Java VM $JAVA failed (see config.log, check the CLASSPATH?))
+fi
+rm -fr $JAVA_TEST $CLASS_TEST Test.uue
+])
+AC_PROVIDE([$0])dnl
+]
+)
+dnl Available from the GNU Autoconf Macro Archive at:
+dnl http://www.gnu.org/software/ac-archive/htmldoc/ac_try_compile_java.html
+dnl
+AC_DEFUN([AC_TRY_COMPILE_JAVA],[
+AC_REQUIRE([AC_PROG_JAVAC])dnl
+cat << \EOF > Test.java
+/* [#]line __oline__ "configure" */
+ifelse([$1], , , [import $1;])
+public class Test {
+[$2]
+}
+EOF
+if AC_TRY_COMMAND($JAVAC $JAVACFLAGS Test.java) && test -s Test.class
+then
+dnl Don't remove the temporary files here, so they can be examined.
+  ifelse([$3], , :, [$3])
+else
+  echo "configure: failed program was:" >&AC_FD_CC
+  cat Test.java >&AC_FD_CC
+ifelse([$4], , , [  rm -fr Test*
+  $4
+])dnl
+fi
+rm -fr Test*])
diff -Nru icedtea6-1.2.orig/m4/classpath_javac_mem_check.m4 icedtea6-1.2/m4/classpath_javac_mem_check.m4
--- icedtea6-1.2.orig/m4/classpath_javac_mem_check.m4	1970-01-01 01:00:00.000000000 +0100
+++ icedtea6-1.2/m4/classpath_javac_mem_check.m4	2008-06-29 21:40:14.000000000 +0100
@@ -0,0 +1,27 @@
+dnl -----------------------------------------------------------
+AC_DEFUN([CLASSPATH_JAVAC_MEM_CHECK],
+[
+  JAVA_TEST=Test.java
+  CLASS_TEST=Test.class
+  cat << \EOF > $JAVA_TEST
+  /* [#]line __oline__ "configure" */
+  public class Test 
+  {
+    public static void main(String[] args)
+    {
+      System.out.println("Hello World");
+    }
+  }
+EOF
+  AC_MSG_CHECKING([whether javac supports -J])
+  $JAVAC $JAVACFLAGS -J-Xmx768M -sourcepath '' $JAVA_TEST
+  javac_result=$?
+  if test "x$javac_result" = "x0"; then
+    AC_MSG_RESULT([yes])
+    JAVAC_MEM_OPT="-J-Xmx768M"
+  else
+    AC_MSG_RESULT([no])
+  fi
+  rm -f $JAVA_TEST $CLASS_TEST
+  AC_SUBST(JAVAC_MEM_OPT)
+])
diff -Nru icedtea6-1.2.orig/Makefile.am icedtea6-1.2/Makefile.am
--- icedtea6-1.2.orig/Makefile.am	2008-05-28 14:15:29.000000000 +0100
+++ icedtea6-1.2/Makefile.am	2008-06-29 21:56:07.000000000 +0100
@@ -1,3 +1,6 @@
+# Allow users to pass site-specific flags to autoreconf via an env var. 
+ACLOCAL_AMFLAGS = -I m4 ${LOCAL_AUTORECONF_FLAGS}
+
 OPENJDK_DATE = 11_apr_2008
 OPENJDK_MD5SUM = 9ef5b6e83ec0c7a32bffd1a4de50fe5c
 OPENJDK_VERSION = b09
@@ -555,16 +558,6 @@
 # OpenJDK Targets
 # ===============
 
-if WITH_ICEDTEA
-if WITH_CACAO
-  MEMORY_LIMIT = -J-Xmx1024m
-else
-  MEMORY_LIMIT = -J-Xmx512m
-endif
-else
-  MEMORY_LIMIT =
-endif
-
 if WITH_CACAO
   ICEDTEA_BUILD_TARGET=j2se_only
   ICEDTEA_DEBUG_BUILD_TARGET=j2se_fastdebug_only
@@ -762,28 +755,28 @@
 	  cp -pPRf openjdk/jdk/src/share/classes/$$copy_dir/* \
 	    hotspot-tools/$$copy_dir ; \
 	done
-	
+
 	for copy_dir in `cat tools-copy/tools-langtools-copy-files.txt` ; \
 	do \
 	  mkdir -p hotspot-tools/$$copy_dir ; \
 	  cp -pPRf openjdk/langtools/src/share/classes/$$copy_dir/* \
 	    hotspot-tools/$$copy_dir ; \
 	done
-	
+
 	for copy_dir in `cat tools-copy/tools-corba-copy-files.txt` ; \
 	do \
 	  mkdir -p hotspot-tools/$$copy_dir ; \
 	  cp -pPRf openjdk/corba/src/share/classes/$$copy_dir/* \
 	    hotspot-tools/$$copy_dir ; \
 	done
-	
+
 	for copy_dir in `cat tools-copy/tools-jaxws-copy-files.txt` ; \
 	do \
 	  mkdir -p hotspot-tools/$$copy_dir ; \
 	  cp -pPRf openjdk/jaxws/src/share/classes/$$copy_dir/* \
 	    hotspot-tools/$$copy_dir ; \
 	done
-	
+
 	mkdir -p stamps
 	touch stamps/hotspot-tools-copy-source-files.stamp
 
@@ -795,12 +788,12 @@
 stamps/hotspot-tools-class-files.stamp: hotspot-tools-source-files.txt
 	if ! test -d $(ICEDTEA_BOOT_DIR) ; \
 	then \
-	  $(JAVAC) $(MEMORY_LIMIT) -g -d lib/hotspot-tools \
+	  $(JAVAC) $(JAVAC_MEM_OPT) -g -d lib/hotspot-tools \
             -source 1.5 \
             -sourcepath rt:hotspot-tools:$(OPENJDK_SOURCEPATH_DIRS):generated \
             -bootclasspath '' @$< ; \
 	else \
-	  $(ICEDTEA_BOOT_DIR)/bin/javac $(MEMORY_LIMIT) -g -d lib/hotspot-tools \
+	  $(ICEDTEA_BOOT_DIR)/bin/javac $(JAVAC_MEM_OPT) -g -d lib/hotspot-tools \
 	    -source 1.5 \
 	    -sourcepath rt:hotspot-tools:$(OPENJDK_SOURCEPATH_DIRS):generated \
 	    -bootclasspath '' @$< ; \
@@ -842,12 +835,12 @@
 	mkdir -p lib/rt
 	if ! test -d $(ICEDTEA_BOOT_DIR) ; \
 	then \
-	  $(JAVAC) $(MEMORY_LIMIT) -g -d lib/rt \
+	  $(JAVAC) $(JAVAC_MEM_OPT) -g -d lib/rt \
 	    -source 1.5 \
 	    -sourcepath rt:$(OPENJDK_SOURCEPATH_DIRS):generated \
 	    -bootclasspath '' @$< ; \
 	else \
-	  $(ICEDTEA_BOOT_DIR)/bin/javac $(MEMORY_LIMIT) -g -d lib/rt \
+	  $(ICEDTEA_BOOT_DIR)/bin/javac $(JAVAC_MEM_OPT) -g -d lib/rt \
             -source 1.5 \
             -sourcepath rt:$(OPENJDK_SOURCEPATH_DIRS):generated \
             -bootclasspath '' @$< ; \
@@ -887,12 +880,12 @@
 	mkdir -p extra-lib
 	if ! test -d $(ICEDTEA_BOOT_DIR) ; \
 	then \
-	  $(JAVAC) $(MEMORY_LIMIT) -g -d extra-lib \
+	  $(JAVAC) $(JAVAC_MEM_OPT) -g -d extra-lib \
 	    -source 1.5 \
- 	    -sourcepath extra -cp bootstrap/jdk1.7.0/jre/lib/rt-closed.jar \
+	    -sourcepath extra -cp bootstrap/jdk1.7.0/jre/lib/rt-closed.jar \
 	    -bootclasspath '' @extra-source-files.txt ; \
 	else \
-	  $(ICEDTEA_BOOT_DIR)/bin/javac $(MEMORY_LIMIT) -g -d extra-lib \
+	  $(ICEDTEA_BOOT_DIR)/bin/javac $(JAVAC_MEM_OPT) -g -d extra-lib \
             -source 1.5 \
             -sourcepath extra -cp bootstrap/jdk1.7.0/jre/lib/rt-closed.jar \
             -bootclasspath '' @extra-source-files.txt ; \
