diff -Nru icedtea6-1.2.orig/configure.ac icedtea6-1.2/configure.ac
--- icedtea6-1.2.orig/configure.ac	2008-07-20 22:14:54.000000000 +0100
+++ icedtea6-1.2/configure.ac	2008-07-20 22:15:06.000000000 +0100
@@ -216,9 +216,6 @@
 AC_CHECK_HEADERS([X11/X.h],[]
 	,[AC_MSG_ERROR("xorg headers were not found - 
 	try installing xorg-x11-proto-devel.")])
-AC_CHECK_HEADERS([Xm/Xm.h Xm/XmP.h Xm/Display.h],[]
-	,[AC_MSG_ERROR("motif headers were not found - 
-	try installing lesstif-devel.")])
 AC_CHECK_LIB(jpeg, main,
         , [AC_MSG_ERROR("libjpeg not found - try installing libjpeg-devel")])
 AC_CHECK_LIB(gif, main,
