diff -Nru icedtea6-1.2.orig/Makefile.am icedtea6-1.2/Makefile.am
--- icedtea6-1.2.orig/Makefile.am	2008-07-20 22:18:12.000000000 +0100
+++ icedtea6-1.2/Makefile.am	2008-07-20 22:18:52.000000000 +0100
@@ -318,6 +318,8 @@
 	patches/icedtea-hat-spl-gpl.patch \
 	patches/icedtea-sparc.patch \
 	patches/icedtea-override-redirect-metacity.patch \
+	patches/icedtea-nomotif-6706121.patch \
+        patches/icedtea-nomotif.patch \
 	$(GCC_PATCH) \
 	$(DISTRIBUTION_PATCHES)
 
