diff -ru icedtea6-1.1/acinclude.m4 icedtea6-1.1-new/acinclude.m4
--- icedtea6-1.1/acinclude.m4	2008-04-03 19:03:28.000000000 +0100
+++ icedtea6-1.1-new/acinclude.m4	2008-04-18 23:37:22.000000000 +0100
@@ -521,14 +521,26 @@
                 [AS_HELP_STRING(--enable-zero,
                                use zero-assembler port on non-zero platforms)],
   [
-    use_zero=yes
+    case "${enableval}" in
+      no)
+        use_zero=no
+        ;;
+      *)
+        use_zero=yes
+        ;;
+    esac
   ],
   [
     case "${host}" in
       i?86-*-*) ;;
       x86_64-*-*) ;;
       *)
-        use_zero=yes
+        if test "x${CACAO}" != xno; then
+          use_zero=no
+        else
+          use_zero=yes
+        fi
+        ;;
     esac
   ])
   AC_MSG_RESULT($use_zero)
@@ -543,10 +555,10 @@
     ZERO_LIBARCH="${INSTALL_ARCH_DIR}"
     dnl can't use AC_CHECK_SIZEOF on multilib
     case "${ZERO_LIBARCH}" in
-      i386|ppc|s390)
+      i386|ppc|s390|sparc)
         ZERO_BITSPERWORD=32
         ;;
-      amd64|ppc64|s390x)
+      amd64|ppc64|s390x|sparc64)
         ZERO_BITSPERWORD=64
         ;;
       *)
@@ -564,18 +576,21 @@
       s390*)
         ZERO_ARCHDEF="S390"
         ;;
+      sparc*)
+        ZERO_ARCHDEF="SPARC"
+        ;;
       *)
         ZERO_ARCHDEF=`echo ${ZERO_LIBARCH} | tr a-z A-Z`
     esac
     dnl multilib machines need telling which mode to build for
     case "${ZERO_LIBARCH}" in
-      i386|ppc)
+      i386|ppc|sparc)
         ZERO_ARCHFLAG="-m32"
         ;;
       s390)
         ZERO_ARCHFLAG="-m31"
         ;;
-      amd64|ppc64|s390x)
+      amd64|ppc64|s390x|sparc64)
         ZERO_ARCHFLAG="-m64"
         ;;
     esac
