*** build.xml.orig	Wed Nov 10 14:45:16 2004
--- build.xml	Wed Nov 10 15:14:19 2004
***************
*** 25,34 ****
    <target name="init" description="o Initializes some properties">
      <mkdir dir="${libdir}">
      </mkdir>
!     <condition property="noget">
        <equals arg2="only" arg1="${build.sysclasspath}">
        </equals>
!     </condition>
    </target>
    <target name="compile" description="o Compile the code" depends="get-deps">
      <mkdir dir="${classesdir}">
--- 25,35 ----
    <target name="init" description="o Initializes some properties">
      <mkdir dir="${libdir}">
      </mkdir>
!     <!--<condition property="noget">
        <equals arg2="only" arg1="${build.sysclasspath}">
        </equals>
!     </condition>-->
!     <property name="noget" value="true"></property>
    </target>
    <target name="compile" description="o Compile the code" depends="get-deps">
      <mkdir dir="${classesdir}">
***************
*** 172,175 ****
      <unjar dest="${maven.home}" src="${user.home}/maven-install-latest.jar">
      </unjar>
    </target>
! </project>
\ No newline at end of file
--- 173,176 ----
      <unjar dest="${maven.home}" src="${user.home}/maven-install-latest.jar">
      </unjar>
    </target>
! </project>
