--- org/gudy/azureus2/platform/PlatformManagerFactory.java.orig	2008-07-06 22:58:18.000000000 +0200
+++ org/gudy/azureus2/platform/PlatformManagerFactory.java	2008-07-06 22:59:21.000000000 +0200
@@ -50,10 +50,6 @@
 						
 						platform_manager = org.gudy.azureus2.platform.win32.PlatformManagerImpl.getSingleton();
 						
-					}else if( getPlatformType() == PlatformManager.PT_MACOSX ){
-						
-	                    platform_manager = org.gudy.azureus2.platform.macosx.PlatformManagerImpl.getSingleton();
-	                    
 					}else if( getPlatformType() == PlatformManager.PT_UNIX ){
 						
             platform_manager = org.gudy.azureus2.platform.unix.PlatformManagerImpl.getSingleton();
@@ -89,10 +85,6 @@
 
 			return (PlatformManager.PT_WINDOWS );
 
-		} else if (Constants.isOSX) {
-
-			return (PlatformManager.PT_MACOSX );
-
 		} else if (Constants.isUnix) {
 
 			return (PlatformManager.PT_UNIX );
