--- net/sf/sevenzipjbinding/SevenZip.java.orig	2010-03-06 21:48:58.000000000 +0000
+++ net/sf/sevenzipjbinding/SevenZip.java	2011-01-02 20:01:57.393194867 +0000
@@ -334,6 +334,10 @@
 				return;
 			}
 
+			System.loadLibrary("7-Zip-JBinding");
+			nativeInitialization();
+			if (true) { return; }
+
 			String pathInJAR = platform;
 			if (pathInJAR == null) {
 				pathInJAR = getPlatformBestMatch();
