diff -Nru resin-3.0.19.vanilla/Makefile.in resin-3.0.19/Makefile.in
--- resin-3.0.19.vanilla/Makefile.in	2006-05-27 17:02:08.000000000 +0200
+++ resin-3.0.19/Makefile.in	2006-05-27 17:03:04.000000000 +0200
@@ -8,20 +8,20 @@
 
 install	:
 	(cd modules/c/src; $(MAKE) install)
-	if test $(PREFIX) != `pwd`; then \
-	  mkdir -p $(PREFIX)/lib; \
-	  mkdir -p $(PREFIX)/libexec; \
-	  cp -r libexec/* $(PREFIX)/libexec; \
-	  cp lib/*.jar $(PREFIX)/lib; \
-	  mkdir -p $(PREFIX)/bin; \
-	  cp bin/* $(PREFIX)/bin; \
-	  mkdir -p $(PREFIX)/webapps; \
-	  cp -r webapps/* $(PREFIX)/webapps; \
-	  mkdir -p $(PREFIX)/conf; \
-	  cp conf/resin.conf $(PREFIX)/conf/resin.conf.orig; \
-	  cp conf/app-default.xml $(PREFIX)/conf/app-default.xml.orig; \
-	  if test ! -r $(PREFIX)/conf/resin.conf; then \
-	    cp conf/resin.conf $(PREFIX)/conf/resin.conf; \
-	    cp conf/app-default.xml $(PREFIX)/conf/app-default.xml; \
+	if test $(DESTDIR)$(PREFIX) != `pwd`; then \
+	  mkdir -p $(DESTDIR)$(PREFIX)/lib; \
+	  mkdir -p $(DESTDIR)$(PREFIX)/libexec; \
+	  cp -r libexec/* $(DESTDIR)$(PREFIX)/libexec; \
+	  cp lib/*.jar $(DESTDIR)$(PREFIX)/lib; \
+	  mkdir -p $(DESTDIR)$(PREFIX)/bin; \
+	  cp bin/* $(DESTDIR)$(PREFIX)/bin; \
+	  mkdir -p $(DESTDIR)$(PREFIX)/webapps; \
+	  cp -r webapps/* $(DESTDIR)$(PREFIX)/webapps; \
+	  mkdir -p $(DESTDIR)$(PREFIX)/conf; \
+	  cp conf/resin.conf $(DESTDIR)$(PREFIX)/conf/resin.conf.orig; \
+	  cp conf/app-default.xml $(DESTDIR)$(PREFIX)/conf/app-default.xml.orig; \
+	  if test ! -r $(DESTDIR)$(PREFIX)/conf/resin.conf; then \
+	    cp conf/resin.conf $(DESTDIR)$(PREFIX)/conf/resin.conf; \
+	    cp conf/app-default.xml $(DESTDIR)$(PREFIX)/conf/app-default.xml; \
 	  fi; \
 	fi
diff -Nru resin-3.0.19.vanilla/bin/wrapper.pl resin-3.0.19/bin/wrapper.pl
--- resin-3.0.19.vanilla/bin/wrapper.pl	2006-05-27 17:01:59.000000000 +0200
+++ resin-3.0.19/bin/wrapper.pl	2006-05-27 17:03:34.000000000 +0200
@@ -441,9 +441,6 @@
 	$pid=`cat $pid_file`;
 	chop($pid);
 
-	if ($cmd eq "stop") {
-	    print("Stopping ${name}\n");
-	}
 	kill(15, $pid);
 	unlink($pid_file);
     }
@@ -729,8 +726,6 @@
       $date = `date`;
       chop($date);
 
-      print "Resin $name $cmd at $date\n";
-
       # create a keepalive socket
       # when the wrapper dies, the httpd class will detect that and
       # close gracefully
diff -Nru resin-3.0.19.vanilla/bin/wrapper.pl.in resin-3.0.19/bin/wrapper.pl.in
--- resin-3.0.19.vanilla/bin/wrapper.pl.in	2006-05-27 17:01:59.000000000 +0200
+++ resin-3.0.19/bin/wrapper.pl.in	2006-05-27 17:03:52.000000000 +0200
@@ -441,9 +441,6 @@
 	$pid=`cat $pid_file`;
 	chop($pid);
 
-	if ($cmd eq "stop") {
-	    print("Stopping ${name}\n");
-	}
 	kill(15, $pid);
 	unlink($pid_file);
     }
@@ -729,8 +726,6 @@
       $date = `date`;
       chop($date);
 
-      print "Resin $name $cmd at $date\n";
-
       # create a keepalive socket
       # when the wrapper dies, the httpd class will detect that and
       # close gracefully
diff -Nru resin-3.0.19.vanilla/build.xml resin-3.0.19/build.xml
--- resin-3.0.19.vanilla/build.xml	2006-05-27 17:02:08.000000000 +0200
+++ resin-3.0.19/build.xml	2006-05-27 17:04:57.000000000 +0200
@@ -7,7 +7,6 @@
 
   <property name="modules" location="${basedir}/modules"/>
   <property name="src" location="${basedir}/src"/>
-  <property name="ext" location="${basedir}/modules/ext"/>
   <property name="doc" location="${basedir}/doc"/>
 
   <property name="build" location="${install}/build"/>
@@ -66,7 +65,7 @@
   </target>
 
   <target name="compile" 
-          depends="init, ejb, ejb30, j2eedeploy, jaxrpc, jca, jms, jmx, jsdk, jstl, jta, mail, portlet, script, resin-jdk15, resin, quercus, webutil, deploy, conf, ext">
+          depends="init, ejb, ejb30, j2eedeploy, jaxrpc, jca, jms, jmx, jsdk, jstl, jta, mail, portlet, script, resin-jdk15, resin, quercus, webutil, deploy, conf">
   </target>
 
   <target name="ejb" depends="init, jta">
@@ -266,9 +265,6 @@
         <dirset dir="${modules}">
           <include name="*/classes"/>
        </dirset>
-        <fileset dir="${ext}">
-          <include name="**/*.jar"/>
-        </fileset>
       </classpath>
     </javac>
 
@@ -331,13 +327,6 @@
     </copy>
   </target>
 
-  <target name="ext" depends="init, resin">
-    <copy todir="${install}/lib" preservelastmodified="true">
-      <fileset dir="${ext}">
-      </fileset>
-    </copy>
-  </target>
-
   <target name="webapps" depends="init, doc, quercus">
   </target>
 
diff -Nru resin-3.0.19.vanilla/modules/resin/src/com/caucho/server/resin/Resin.java resin-3.0.19/modules/resin/src/com/caucho/server/resin/Resin.java
--- resin-3.0.19.vanilla/modules/resin/src/com/caucho/server/resin/Resin.java	2006-05-27 17:02:07.000000000 +0200
+++ resin-3.0.19/modules/resin/src/com/caucho/server/resin/Resin.java	2006-05-27 17:05:48.000000000 +0200
@@ -699,10 +699,10 @@
       throw new ConfigException(L().l("The following system property must be set:\n  -Djava.util.logging.manager=com.caucho.log.LogManagerImpl\nThe JDK 1.4 Logging manager must be set to Resin's log manager."));
     }
 
-    validatePackage("javax.servlet.Servlet", new String[] {"2.4", "1.4"});
+    validatePackage("javax.servlet.Servlet", new String[] {"2.4", "1.5"});
     validatePackage("javax.servlet.jsp.jstl.core.Config", new String[] {"1.1"});
     validatePackage("javax.management.MBeanServer", new String[] { "1.2", "1.5" });
-    validatePackage("javax.resource.spi.ResourceAdapter", new String[] {"1.5", "1.4"});
+    validatePackage("javax.resource.spi.ResourceAdapter", new String[] {"1.5", "1.5"});
   }
 
   /**
