--- cairo.c	2006-07-17 16:02:31.000000000 +0200
+++ cairo.c	2006-07-17 16:02:46.000000000 +0200
@@ -533,7 +533,7 @@
 {
 	jlong rc = 0;
 	Cairo_NATIVE_ENTER(env, that, cairo_1image_1surface_1create_1for_1data_FUNC);
-	rc = (jlong)cairo_image_surface_create_for_data((char *)arg0, arg1, arg2, arg3, arg4);
+	rc = (jlong)cairo_image_surface_create_for_data((unsigned char *)arg0, arg1, arg2, arg3, arg4);
 	Cairo_NATIVE_EXIT(env, that, cairo_1image_1surface_1create_1for_1data_FUNC);
 	return rc;
 }
