--- cairo.c.orig	2005-08-11 15:44:50.000000000 +0000
+++ cairo.c	2005-09-10 21:45:59.747679632 +0000
@@ -254,6 +254,62 @@
 }
 #endif
 
+#ifndef NO_cairo_1font_1options_1create
+JNIEXPORT jlong JNICALL Cairo_NATIVE(cairo_1font_1options_1create)
+	(JNIEnv *env, jclass that)
+{
+	jlong rc = 0;
+	Cairo_NATIVE_ENTER(env, that, cairo_1font_1options_1create_FUNC);
+	rc = (jlong)cairo_font_options_create();
+	Cairo_NATIVE_EXIT(env, that, cairo_1font_1options_1create_FUNC);
+	return rc;
+}
+#endif
+
+#ifndef NO_cairo_1font_1options_1destroy
+JNIEXPORT void JNICALL Cairo_NATIVE(cairo_1font_1options_1destroy)
+	(JNIEnv *env, jclass that, jint arg0)
+{
+	Cairo_NATIVE_ENTER(env, that, cairo_1font_1options_1destroy_FUNC);
+	cairo_font_options_destroy((cairo_font_options_t *)arg0);
+	Cairo_NATIVE_EXIT(env, that, cairo_1font_1options_1destroy_FUNC);
+}
+#endif
+
+#ifndef NO_cairo_1font_1options_1get_1antialias
+JNIEXPORT jlong JNICALL Cairo_NATIVE(cairo_1font_1options_1get_1antialias)
+	(JNIEnv *env, jclass that, jlong arg0)
+{
+	jlong rc = 0;
+	Cairo_NATIVE_ENTER(env, that, cairo_1font_1options_1get_1antialias_FUNC);
+	rc = (jlong)cairo_font_options_get_antialias((cairo_font_options_t *)arg0);
+	Cairo_NATIVE_EXIT(env, that, cairo_1font_1options_1get_1antialias_FUNC);
+	return rc;
+}
+#endif
+
+#ifndef NO_cairo_1font_1options_1set_1antialias
+JNIEXPORT void JNICALL Cairo_NATIVE(cairo_1font_1options_1set_1antialias)
+	(JNIEnv *env, jclass that, jlong arg0, jlong arg1)
+{
+	Cairo_NATIVE_ENTER(env, that, cairo_1font_1options_1set_1antialias_FUNC);
+	cairo_font_options_set_antialias((cairo_font_options_t *)arg0, (cairo_antialias_t)arg1);
+	Cairo_NATIVE_EXIT(env, that, cairo_1font_1options_1set_1antialias_FUNC);
+}
+#endif
+
+#ifndef NO_cairo_1get_1antialias
+JNIEXPORT jlong JNICALL Cairo_NATIVE(cairo_1get_1antialias)
+	(JNIEnv *env, jclass that, jlong arg0)
+{
+	jlong rc = 0;
+	Cairo_NATIVE_ENTER(env, that, cairo_1get_1antialias_FUNC);
+	rc = (jlong)cairo_get_antialias((cairo_t *)arg0);
+	Cairo_NATIVE_EXIT(env, that, cairo_1get_1antialias_FUNC);
+	return rc;
+}
+#endif
+
 #ifndef NO_cairo_1get_1current_1point
 JNIEXPORT void JNICALL Cairo_NATIVE(cairo_1get_1current_1point)
 	(JNIEnv *env, jclass that, jlong arg0, jdoubleArray arg1, jdoubleArray arg2)
@@ -972,12 +1028,14 @@
 #endif
 
 #ifndef NO_cairo_1reference
-JNIEXPORT void JNICALL Cairo_NATIVE(cairo_1reference)
+JNIEXPORT jlong JNICALL Cairo_NATIVE(cairo_1reference)
 	(JNIEnv *env, jclass that, jlong arg0)
 {
+	jlong rc = 0;
 	Cairo_NATIVE_ENTER(env, that, cairo_1reference_FUNC);
-	cairo_reference((cairo_t *)arg0);
+	rc = (jlong)cairo_reference((cairo_t *)arg0);
 	Cairo_NATIVE_EXIT(env, that, cairo_1reference_FUNC);
+	return rc;
 }
 #endif
 
@@ -1075,6 +1133,16 @@
 }
 #endif
 
+#ifndef NO_cairo_1set_1antialias
+JNIEXPORT void JNICALL Cairo_NATIVE(cairo_1set_1antialias)
+	(JNIEnv *env, jclass that, jlong arg0, jlong arg1)
+{
+	Cairo_NATIVE_ENTER(env, that, cairo_1set_1antialias_FUNC);
+	cairo_set_antialias((cairo_t *)arg0, (cairo_antialias_t)arg1);
+	Cairo_NATIVE_EXIT(env, that, cairo_1set_1antialias_FUNC);
+}
+#endif
+
 #ifndef NO_cairo_1set_1dash
 JNIEXPORT void JNICALL Cairo_NATIVE(cairo_1set_1dash)
 	(JNIEnv *env, jclass that, jlong arg0, jdoubleArray arg1, jint arg2, jdouble arg3)
@@ -1123,6 +1191,16 @@
 }
 #endif
 
+#ifndef NO_cairo_1set_1font_1options
+JNIEXPORT void JNICALL Cairo_NATIVE(cairo_1set_1font_1options)
+	(JNIEnv *env, jclass that, jlong arg0, jlong arg1)
+{
+	Cairo_NATIVE_ENTER(env, that, cairo_1set_1font_1options_FUNC);
+	cairo_set_font_options((cairo_t *)arg0, (cairo_font_options_t *)arg1);
+	Cairo_NATIVE_EXIT(env, that, cairo_1set_1font_1options_FUNC);
+}
+#endif
+
 #ifndef NO_cairo_1set_1font_1size
 JNIEXPORT void JNICALL Cairo_NATIVE(cairo_1set_1font_1size)
 	(JNIEnv *env, jclass that, jlong arg0, jdouble arg1)
@@ -1133,6 +1211,16 @@
 }
 #endif
 
+#ifndef NO_cairo_1get_1font_1options
+JNIEXPORT void JNICALL Cairo_NATIVE(cairo_1get_1font_1options)
+	(JNIEnv *env, jclass that, jlong arg0, jlong arg1)
+{
+	Cairo_NATIVE_ENTER(env, that, cairo_1get_1font_1options_FUNC);
+	cairo_get_font_options((cairo_t *)arg0, (cairo_font_options_t *)arg1);
+	Cairo_NATIVE_EXIT(env, that, cairo_1get_1font_1options_FUNC);
+}
+#endif
+
 #ifndef NO_cairo_1set_1line_1cap
 JNIEXPORT void JNICALL Cairo_NATIVE(cairo_1set_1line_1cap)
 	(JNIEnv *env, jclass that, jlong arg0, jint arg1)
@@ -1371,14 +1459,12 @@
 #endif
 
 #ifndef NO_cairo_1surface_1finish
-JNIEXPORT jint JNICALL Cairo_NATIVE(cairo_1surface_1finish)
+JNIEXPORT void JNICALL Cairo_NATIVE(cairo_1surface_1finish)
 	(JNIEnv *env, jclass that, jlong arg0)
 {
-	jint rc = 0;
 	Cairo_NATIVE_ENTER(env, that, cairo_1surface_1finish_FUNC);
-	rc = (jint)cairo_surface_finish((cairo_surface_t *)arg0);
+	cairo_surface_finish((cairo_surface_t *)arg0);
 	Cairo_NATIVE_EXIT(env, that, cairo_1surface_1finish_FUNC);
-	return rc;
 }
 #endif
 
@@ -1526,11 +1612,11 @@
 
 #ifndef NO_cairo_1xlib_1surface_1create_1for_1bitmap
 JNIEXPORT jlong JNICALL Cairo_NATIVE(cairo_1xlib_1surface_1create_1for_1bitmap)
-	(JNIEnv *env, jclass that, jlong arg0, jlong arg1, jint arg2, jint arg3)
+	(JNIEnv *env, jclass that, jlong arg0, jlong arg1, jlong arg2, jint arg3, jint arg4)
 {
 	jlong rc = 0;
 	Cairo_NATIVE_ENTER(env, that, cairo_1xlib_1surface_1create_1for_1bitmap_FUNC);
-	rc = (jlong)cairo_xlib_surface_create_for_bitmap((Display *)arg0, (Pixmap)arg1, arg2, arg3);
+	rc = (jlong)cairo_xlib_surface_create_for_bitmap((Display *)arg0, (Pixmap)arg1, (Screen *)arg2, arg3, arg4);
 	Cairo_NATIVE_EXIT(env, that, cairo_1xlib_1surface_1create_1for_1bitmap_FUNC);
 	return rc;
 }

===================================================================
RCS file: /home/eclipse/org.eclipse.swt/Eclipse SWT PI/cairo/library/cairo_stats.c,v
retrieving revision 1.6
retrieving revision 1.7
diff -u -r1.6 -r1.7
--- cairo_stats.c	2005/07/11 21:28:17	1.6
+++ cairo_stats.c	2005/08/15 20:22:55	1.7
@@ -24,8 +24,8 @@
 
 #ifdef NATIVE_STATS
 
-int Cairo_nativeFunctionCount = 130;
-int Cairo_nativeFunctionCallCount[130];
+int Cairo_nativeFunctionCount = 138;
+int Cairo_nativeFunctionCallCount[138];
 char * Cairo_nativeFunctionNames[] = {
 	"cairo_1append_1path",
 	"cairo_1arc",
@@ -46,10 +46,16 @@
 	"cairo_1fill_1preserve",
 	"cairo_1font_1extents",
 	"cairo_1font_1extents_1t_1sizeof",
+	"cairo_1font_1options_1create",
+	"cairo_1font_1options_1destroy",
+	"cairo_1font_1options_1get_1antialias",
+	"cairo_1font_1options_1set_1antialias",
+	"cairo_1get_1antialias",
 	"cairo_1get_1current_1point",
 	"cairo_1get_1fill_1rule",
 	"cairo_1get_1font_1face",
 	"cairo_1get_1font_1matrix",
+	"cairo_1get_1font_1options",
 	"cairo_1get_1line_1cap",
 	"cairo_1get_1line_1join",
 	"cairo_1get_1line_1width",
@@ -114,10 +120,12 @@
 	"cairo_1save",
 	"cairo_1scale",
 	"cairo_1select_1font_1face",
+	"cairo_1set_1antialias",
 	"cairo_1set_1dash",
 	"cairo_1set_1fill_1rule",
 	"cairo_1set_1font_1face",
 	"cairo_1set_1font_1matrix",
+	"cairo_1set_1font_1options",
 	"cairo_1set_1font_1size",
 	"cairo_1set_1line_1cap",
 	"cairo_1set_1line_1join",

===================================================================
RCS file: /home/eclipse/org.eclipse.swt/Eclipse SWT PI/cairo/library/cairo_stats.h,v
retrieving revision 1.6
retrieving revision 1.7
diff -u -r1.6 -r1.7
--- cairo_stats.h	2005/07/11 21:28:17	1.6
+++ cairo_stats.h	2005/08/15 20:22:55	1.7
@@ -50,10 +50,16 @@
 	cairo_1fill_1preserve_FUNC,
 	cairo_1font_1extents_FUNC,
 	cairo_1font_1extents_1t_1sizeof_FUNC,
+	cairo_1font_1options_1create_FUNC,
+	cairo_1font_1options_1destroy_FUNC,
+	cairo_1font_1options_1get_1antialias_FUNC,
+	cairo_1font_1options_1set_1antialias_FUNC,
+	cairo_1get_1antialias_FUNC,
 	cairo_1get_1current_1point_FUNC,
 	cairo_1get_1fill_1rule_FUNC,
 	cairo_1get_1font_1face_FUNC,
 	cairo_1get_1font_1matrix_FUNC,
+	cairo_1get_1font_1options_FUNC,
 	cairo_1get_1line_1cap_FUNC,
 	cairo_1get_1line_1join_FUNC,
 	cairo_1get_1line_1width_FUNC,
@@ -118,10 +124,12 @@
 	cairo_1save_FUNC,
 	cairo_1scale_FUNC,
 	cairo_1select_1font_1face_FUNC,
+	cairo_1set_1antialias_FUNC,
 	cairo_1set_1dash_FUNC,
 	cairo_1set_1fill_1rule_FUNC,
 	cairo_1set_1font_1face_FUNC,
 	cairo_1set_1font_1matrix_FUNC,
+	cairo_1set_1font_1options_FUNC,
 	cairo_1set_1font_1size_FUNC,
 	cairo_1set_1line_1cap_FUNC,
 	cairo_1set_1line_1join_FUNC,
