diff -ur eclipse.orig/plugins/org.eclipse.swt/Eclipse SWT Mozilla/common/library/xpcom.cpp eclipse/plugins/org.eclipse.swt/Eclipse SWT Mozilla/common/library/xpcom.cpp
--- eclipse.orig/plugins/org.eclipse.swt/Eclipse SWT Mozilla/common/library/xpcom.cpp	2005-03-11 15:59:28.000000000 +0200
+++ eclipse/plugins/org.eclipse.swt/Eclipse SWT Mozilla/common/library/xpcom.cpp	2005-03-30 08:03:03.983042167 +0300
@@ -234,16 +234,6 @@
 	return rc;
 }
 
-JNIEXPORT jboolean JNICALL XPCOM_NATIVE(nsString_1Equals)
-	(JNIEnv *env, jclass, jint arg0, jint arg1)
-{
-	nsEmbedString *lparg0 = NULL;
-	nsEmbedString *lparg1 = NULL;
-	if (arg0 != 0) lparg0 = (nsEmbedString*)arg0;
-	if (arg1 != 0) lparg1 = (nsEmbedString*)arg1;
-	return lparg0->Equals(*lparg1);
-}
-
 JNIEXPORT jstring JNICALL XPCOM_NATIVE(PR_1GetEnv)
 	(JNIEnv *env, jclass, jstring arg0)
 {
