Index: batch/org/eclipse/jdt/internal/compiler/batch/Main.java
===================================================================
RCS file: /cvsroot/eclipse/org.eclipse.jdt.core/batch/org/eclipse/jdt/internal/compiler/batch/Main.java,v
retrieving revision 1.254
diff -u -r1.254 Main.java
--- batch/org/eclipse/jdt/internal/compiler/batch/Main.java	2 May 2006 13:45:31 -0000	1.254
+++ batch/org/eclipse/jdt/internal/compiler/batch/Main.java	1 Jun 2006 17:56:02 -0000
@@ -2972,7 +2972,7 @@
 	String currentClasspathName = null;
 	ArrayList currentRuleSpecs = new ArrayList(defaultSize);
 	StringTokenizer tokenizer = new StringTokenizer(currentPath,
-			File.pathSeparator + "[]", true); //$NON-NLS-1$
+			File.pathSeparator, true);
 	// state machine
 	final int start = 0; 
 	final int readyToClose = 1;

