--- JspxSerializer.java.orig	2008-11-26 19:34:52.000000000 +0100
+++ JspxSerializer.java	2008-11-26 19:36:19.000000000 +0100
@@ -1381,7 +1381,7 @@
             int ch = source.charAt(i);
             if (!XMLChar.isValid(ch)) {
                 if (++i < length) {
-                    surrogates(ch, source.charAt(i));
+                    surrogates(ch, source.charAt(i), true);
                 } else {
                     fatalError("The character '" + (char) ch + "' is an invalid XML character");
                 }
@@ -1441,7 +1441,7 @@
                 if (!XMLChar.isValid(ch)) {
                     // check if it is surrogate
                     if (++index <length) {
-                        surrogates(ch, text.charAt(index));
+                        surrogates(ch, text.charAt(index), true);
                     } else {
                         fatalError("The character '"+(char)ch+"' is an invalid XML character"); 
                     }
@@ -1463,7 +1463,7 @@
                 if (!XMLChar.isValid(ch)) {
                     // check if it is surrogate
                     if (++index <length) {
-                        surrogates(ch, text.charAt(index));
+                        surrogates(ch, text.charAt(index), true);
                     } else {
                         fatalError("The character '"+(char)ch+"' is an invalid XML character"); 
                     }
@@ -1495,7 +1495,7 @@
                 if (!XMLChar.isValid(ch)) {
                     // check if it is surrogate
                     if (++start <length) {
-                        surrogates(ch, chars[start]);
+                        surrogates(ch, chars[start], true);
                     } else {
                         fatalError("The character '"+(char)ch+"' is an invalid XML character"); 
                     }
@@ -1519,7 +1519,7 @@
                 if (!XMLChar.isValid(ch)) {
                     // check if it is surrogate
                     if (++start <length) {
-                        surrogates(ch, chars[start]);
+                        surrogates(ch, chars[start], true);
                     } else {
                         fatalError("The character '"+(char)ch+"' is an invalid XML character"); 
                     }
