Index: maven-plugins-2.1_pre20071021/maven-javadoc-plugin/src/main/java/org/apache/maven/plugin/javadoc/JavadocReport.java
===================================================================
--- maven-plugins-2.1_pre20071021/maven-javadoc-plugin/src/main/java/org/apache/maven/plugin/javadoc/JavadocReport.java	(révision 586908)
+++ maven-plugins-2.1_pre20071021/maven-javadoc-plugin/src/main/java/org/apache/maven/plugin/javadoc/JavadocReport.java	(copie de travail)
@@ -34,6 +34,9 @@
 import java.util.List;
 import java.util.Locale;
 
+// gentoo-java patch
+import org.apache.maven.doxia.module.xhtml.decoration.render.RenderingContext;
+
 /**
  * Generates documentation for the <code>Java code</code> in the project using the standard
  * <a href="http://java.sun.com/j2se/javadoc/">Javadoc Tool</a>.
@@ -218,10 +221,14 @@
     {
         try
         {
-            SiteRendererSink sink = siteRenderer.createSink( getReportOutputDirectory(), getOutputName() + ".html" );
+            // gentoo-java patch
+            // create sink doesnt exists anymore
+            // siteRenderer.createSink( getReportOutputDirectory(), getOutputName() + ".html" );
+            org.codehaus.doxia.sink.Sink sink = (org.codehaus.doxia.sink.Sink)  new SiteRendererSink( new RenderingContext(getReportOutputDirectory(), getOutputName() + ".html") );
 
             generate( sink, Locale.getDefault() );
         }
+        /* gentoo-java
         catch ( RendererException e )
         {
             throw new MojoExecutionException( "An error has occurred in " + getName( Locale.ENGLISH )
@@ -232,6 +239,7 @@
             throw new MojoExecutionException( "An error has occurred in " + getName( Locale.ENGLISH )
                 + " report generation:" + e.getMessage(), e );
         }
+        */
         catch ( MavenReportException e )
         {
             throw new MojoExecutionException( "An error has occurred in " + getName( Locale.ENGLISH )
