diff -ru jessie-1.0.0/Makefile.in jessie-1.0.0-patched/Makefile.in
--- jessie-1.0.0/Makefile.in	2004-06-09 00:58:58.000000000 -0400
+++ jessie-1.0.0-patched/Makefile.in	2005-09-04 11:22:27.000000000 -0400
@@ -1857,8 +1857,8 @@
 	rm -rf classes
 
 apidoc: .FORCE
-	-mkdir apidoc
-	javadoc -d apidoc/ -classpath $(CLASSPATH) -sourcepath $(srcdir) -bottom "<div style=\"float: left;\"><em>Verbatim copying and distribution of this documentation is permitted in any medium.</em></div><div style=\"margin-top: 10px; margin-bottom: 10px; float: right; text-align: right;\"<a href=\"http://metastatic.org/source/jessie/\"><img src=\"http://metastatic.org/images/jessie-small.png\" width=\"150\" height=\"52\" alt=\"Jessie: A Free Software Implementation of the JSSE\" style=\"border: none;\" /></a></div>" -windowtitle "Jessie API documentation, version $(VERSION)" -link http://www.gnu.org/software/gnu-crypto/api/ -link http://java.sun.com/j2se/1.4.2/docs/api/ -breakiterator -group "JSSE API" javax.net:javax.net.ssl:javax.security.cert -group "Jessie Provider" org.metastatic.jessie:org.metastatic.jessie.https javax.net javax.net.ssl javax.security.cert org.metastatic.jessie org.metastatic.jessie.https
+	-mkdir -p apidoc
+	javadoc -d apidoc/ -classpath $(CLASSPATH) -sourcepath $(srcdir) -bottom "<div style=\"float: left;\"><em>Verbatim copying and distribution of this documentation is permitted in any medium.</em></div><div style=\"margin-top: 10px; margin-bottom: 10px; float: right; text-align: right;\"<a href=\"http://metastatic.org/source/jessie/\"><img src=\"http://metastatic.org/images/jessie-small.png\" width=\"150\" height=\"52\" alt=\"Jessie: A Free Software Implementation of the JSSE\" style=\"border: none;\" /></a></div>" -windowtitle "Jessie API documentation, version $(VERSION)" -link http://www.gnu.org/software/gnu-crypto/manual/api/ -link http://java.sun.com/j2se/1.4.2/docs/api/ -breakiterator -group "JSSE API" javax.net:javax.net.ssl:javax.security.cert -group "Jessie Provider" org.metastatic.jessie:org.metastatic.jessie.https javax.net javax.net.ssl javax.security.cert org.metastatic.jessie org.metastatic.jessie.https
 
 gjdoc: gjdoccatalog.xml .FORCE
 	-mkdir -p apidoc-xml apidoc/common
diff -ru jessie-1.0.0/javax/net/SocketFactory.java jessie-1.0.0-patched/javax/net/SocketFactory.java
--- jessie-1.0.0/javax/net/SocketFactory.java	2003-10-15 19:36:40.000000000 -0400
+++ jessie-1.0.0-patched/javax/net/SocketFactory.java	2005-09-04 11:23:18.000000000 -0400
@@ -154,5 +154,5 @@
    * @return A socket connected to <i>host</i> on <i>port</i>.
    * @throws IOException If a network error occurs.
    */
-  public abstract Socket createSocket(InetAddress hast, int port, InetAddress localHost, int localPort) throws IOException;
+  public abstract Socket createSocket(InetAddress host, int port, InetAddress localHost, int localPort) throws IOException;
 }
