diff -ru maven-1.0/src/bin/maven maven-1.0-patched/bin/maven
--- maven-1.0/src/bin/maven	2004-07-13 09:20:50.000000000 -0400
+++ maven-1.0-patched/src/bin/maven	2005-11-21 19:14:56.000000000 -0500
@@ -45,41 +45,51 @@
            ;;
 esac
 
-if [ -z "$MAVEN_HOME" ] ; then
-  # try to find MAVEN
-  if [ -d /opt/maven ] ; then
+  # try a couple ways to find MAVEN_HOME
+
+  if [ -z ${MAVEN_HOME} ] && [ -d /opt/maven ] ; then
     MAVEN_HOME=/opt/maven
   fi
 
-  if [ -d "$HOME/maven" ] ; then
+  if [ -z ${MAVEN_HOME} ] && [ -d "$HOME/maven" ] ; then
     MAVEN_HOME="$HOME/maven"
   fi
 
-  ## resolve links - $0 may be a link to maven's home
-  PRG=$0
-  progname=`basename $0`
-  saveddir=`pwd`
-
-  # need this for relative symlinks
-  cd `dirname $PRG`
-
-  while [ -h "$PRG" ] ; do
-    ls=`ls -ld "$PRG"`
-    link=`expr "$ls" : '.*-> \(.*\)$'`
-    if expr "$link" : '.*/.*' > /dev/null; then
-  PRG="$link"
-    else
-  PRG="`dirname $PRG`/$link"
-    fi
-  done
+  # try to load from env.d file
+  if [ -z ${MAVEN_HOME} ] && [ -f /etc/env.d/25maven ]; then
+  	MAVEN_HOME=$(source /etc/env.d/25maven; echo ${MAVEN_HOME})
+  fi
 
-  MAVEN_HOME=`dirname "$PRG"`/..
+ # try to figure it out from how this script was invoked
+  if [ -z ${MAVEN_HOME} ]; then
+    ## resolve links - $0 may be a link to maven's home
+    PRG=$0
+    progname=`basename $0`
+    saveddir=`pwd`
+
+    # need this for relative symlinks
+    cd `dirname $PRG`
+
+    while [ -h "$PRG" ] ; do
+      ls=`ls -ld "$PRG"`
+      link=`expr "$ls" : '.*-> \(.*\)$'`
+      if expr "$link" : '.*/.*' > /dev/null; then
+        PRG="$link"
+      else
+        PRG="`dirname $PRG`/$link"
+      fi
+    done
+
+    MAVEN_HOME=`dirname "$PRG"`/..
+
+    # make it fully qualified
+    MAVEN_HOME=`cd "$MAVEN_HOME" && pwd`
+    echo "MAVEN_HOME=${MAVEN_HOME}" 1>&2
 
-  # make it fully qualified
-  MAVEN_HOME=`cd "$MAVEN_HOME" && pwd`
+    cd $saveddir
+  fi
 
-  cd $saveddir
-fi
+[ -z "${MAVEN_HOME}" ] && exit "Was unable to determine MAVEN_HOME"
 
 # For Cygwin, ensure paths are in UNIX format before anything is touched
 if $cygwin ; then
