--- src/test/test/memusage/MemUsageTest.java.orig	2004-10-17 13:33:59.504698704 +0000
+++ src/test/test/memusage/MemUsageTest.java	2004-10-17 13:34:16.539109080 +0000
@@ -137,7 +137,7 @@
         );
         il.append(factory.createLoad(Type.OBJECT, 0));
         il.append(new PUSH(cp, INSTANCE_KSIZE * 1000));
-        il.append(factory.createNewArray(Type.BYTE, (short)1));
+        il.append((Instruction)factory.createNewArray(Type.BYTE, (short)1));
         il.append(factory.createFieldAccess(className, "buffer", new ArrayType(Type.BYTE, 1), Constants.PUTFIELD));
         il.append(factory.createReturn(Type.VOID));
         method.setMaxStack();
@@ -150,7 +150,7 @@
         method =
         new MethodGen(Constants.ACC_STATIC, Type.VOID, Type.NO_ARGS, new String[]{}, "<clinit>", className, il, cp);
         il.append(new PUSH(cp, CLASS_KSIZE * 1000));
-        il.append(factory.createNewArray(Type.BYTE, (short)1));
+        il.append((Instruction)factory.createNewArray(Type.BYTE, (short)1));
         il.append(factory.createFieldAccess(className, "sbuffer", new ArrayType(Type.BYTE, 1), Constants.PUTSTATIC));
         il.append(factory.createReturn(Type.VOID));
         method.setMaxStack();
