Index: plexus-component-api-1.0_alpha24/src/main/java/org/codehaus/plexus/component/repository/ComponentDescriptor.java
===================================================================
--- plexus-component-api-1.0_alpha24/src/main/java/org/codehaus/plexus/component/repository/ComponentDescriptor.java	(révision 6548)
+++ plexus-component-api-1.0_alpha24/src/main/java/org/codehaus/plexus/component/repository/ComponentDescriptor.java	(copie de travail)
@@ -82,7 +82,16 @@
     // ----------------------------------------------------------------------
     //  Instance methods
     // ----------------------------------------------------------------------
+    public String getComponentKey()
+    {
+        if ( getRoleHint() != null )
+        {
+            return getRole() + getRoleHint();
+        }
 
+        return getRole();
+    }
+
     /**
      * Returns a human-friendly key, suitable for display.
      * @return a human-friendly key
