diff -Nur libmatthew-java-0.7/Makefile libmatthew-java-0.7_patched/Makefile
--- libmatthew-java-0.7/Makefile	2008-04-18 22:42:13.000000000 +0300
+++ libmatthew-java-0.7_patched/Makefile	2008-04-23 14:03:21.000000000 +0300
@@ -57,12 +57,7 @@
 io-$(IOVER).jar: .classes
 	(cd classes; $(JAR) cf ../$@ cx/ath/matthew/io/*class)
 unix-$(UNIXVER).jar: .classes
-ifeq ($(DEBUG),enable)
-	echo "Class-Path: $(JARDIR)/debug-$(DEBUG).jar" > Manifest
-else
-	echo "Class-Path: " > Manifest
-endif
-	(cd classes; $(JAR) cfm ../$@ ../Manifest cx/ath/matthew/unix/*class)
+	(cd classes; $(JAR) cf ../$@ cx/ath/matthew/unix/*class)
 
 hexdump-$(HEXVER).jar: .classes
 	(cd classes; $(JAR) cf ../$@ cx/ath/matthew/utils/Hexdump.class)
