--- configure.ac	2005-03-24 22:00:40.000000000 +0100
+++ configure.ac	2005-10-28 18:48:16.000000000 +0200
@@ -325,12 +325,16 @@
   if test "x${enable_gtk_cairo}" = xyes
   then
     PKG_CHECK_MODULES(CAIRO, cairo)
-    PKG_CHECK_MODULES(PANGOFT2, pangoft2)
   fi
+  PKG_CHECK_MODULES(PANGOFT2, pangoft2)
+  PKG_CHECK_MODULES(FREETYPE2, freetype2)
+
   AC_SUBST(CAIRO_LIBS)
   AC_SUBST(CAIRO_CFLAGS)
   AC_SUBST(PANGOFT2_LIBS)
   AC_SUBST(PANGOFT2_CFLAGS)
+  AC_SUBST(FREETYPE2_LIBS)
+  AC_SUBST(FREETYPE2_CFLAGS)
 fi
 
 CLASSPATH_WITH_JAVAH
--- native/jni/gtk-peer/Makefile.am	2005-03-24 21:58:00.000000000 +0100
+++ native/jni/gtk-peer/Makefile.am	2005-10-28 18:51:45.000000000 +0200
@@ -53,10 +53,11 @@
 
 libgtkpeer_la_LIBADD = $(top_builddir)/native/jni/classpath/native_state.lo
 
-AM_LDFLAGS = @CLASSPATH_MODULE@ @GTK_LIBS@ @CAIRO_LIBS@ @PANGOFT2_LIBS@ @X_LIBS@
+AM_LDFLAGS = @CLASSPATH_MODULE@ @GTK_LIBS@ @CAIRO_LIBS@ @PANGOFT2_LIBS@ \
+		@FREETYPE2_LIBS@ @X_PRE_LIBS@ @X_LIBS@ @X_EXTRA_LIBS@ -lX11 -lXtst
 AM_CPPFLAGS = @CLASSPATH_INCLUDES@
 
 # Just the WARNING_CFLAGS. We cannot use the strict flags since the gtk
 # headers contain broken prototypes (by design, see gtkitemfactory.h).
 AM_CFLAGS = @WARNING_CFLAGS@ @ERROR_CFLAGS@ \
-            @GTK_CFLAGS@ @CAIRO_CFLAGS@ @PANGOFT2_CFLAGS@
+            @GTK_CFLAGS@ @CAIRO_CFLAGS@ @PANGOFT2_CFLAGS@ @FREETYPE2_CFLAGS@ @X_FLAGS@
