--- /usr/local/tmp/portage/dev-java/dom4j-1.4/work/dom4j-1.4/src/java/org/dom4j/xpath/DefaultXPath.java	2003-04-07 22:51:33.000000000 +0200
+++ DefaultXPath.java	2007-02-18 00:45:36.703185856 +0100
@@ -58,7 +58,7 @@
 
     /** Construct an XPath
      */
-    public DefaultXPath(String text) throws InvalidXPathException {
+    public DefaultXPath(String text) throws org.jaxen.JaxenException {
         this.text = text;
         this.xpath = parse( text );
     }
@@ -308,13 +313,10 @@
         return valueOf( node );
     }
     
-    protected static XPath parse(String text) {        
+    protected static XPath parse(String text) throws JaxenException {        
         try {
             return new Dom4jXPath( text );
         }
-        catch (SAXPathException e) {
-            throw new InvalidXPathException( text, e.getMessage() );
-        }
         catch (RuntimeException e) {
         }
         throw new InvalidXPathException( text );
