--- /usr/local/tmp/portage/dev-java/dom4j-1.4/work/dom4j-1.4/src/java/org/dom4j/DocumentFactory.java	2003-04-07 22:51:31.000000000 +0200
+++ DocumentFactory.java	2007-02-18 01:10:59.726677451 +0100
@@ -184,13 +184,20 @@
       * @return a new <code>XPath</code> instance
       * @throws InvalidXPathException if the XPath expression is invalid
       */
-    public XPath createXPath(String xpathExpression) throws InvalidXPathException {
-        DefaultXPath xpath = new DefaultXPath( xpathExpression );
-        if ( xpathNamespaceURIs != null ) {
-            xpath.setNamespaceURIs( xpathNamespaceURIs );
+    public XPath createXPath(String xpathExpression) {
+	try{
+		DefaultXPath xpath = new DefaultXPath( xpathExpression );
+		if ( xpathNamespaceURIs != null ) {
+		    xpath.setNamespaceURIs( xpathNamespaceURIs );
+		}
+		return xpath;
+	}
+        catch (org.jaxen.JaxenException e) {		
+            System.out.println("error in createXPath(String xpathExpression)");
+	    return null;
         }
-        return xpath;
     }
+    
 
     /** <p><code>createXPath</code> parses an XPath expression
       * and creates a new XPath <code>XPath</code> instance.</p>
@@ -245,7 +252,7 @@
       * to create
       * @return a new <code>Pattern</code> instance
       */
-    public Pattern createPattern(String xpathPattern) {
+    public Pattern createPattern(String xpathPattern) throws org.jaxen.JaxenException {
         return new XPathPattern( xpathPattern );
     }
 
