--- /usr/local/tmp/portage/dev-java/dom4j-1.4/work/dom4j-1.4/src/java/org/dom4j/util/ProxyDocumentFactory.java	2003-04-07 22:51:34.000000000 +0200
+++ ProxyDocumentFactory.java	2007-02-18 01:44:12.589200952 +0100
@@ -147,7 +147,7 @@
         return proxy.createXPathFilter(xpathFilterExpression);
     }
     
-    public Pattern createPattern(String xpathPattern) {
+    public Pattern createPattern(String xpathPattern) throws org.jaxen.JaxenException {
         return proxy.createPattern(xpathPattern);
     }
     
