--- /usr/local/tmp/portage/dev-java/dom4j-1.4/work/dom4j-1.4/src/java/org/dom4j/xpp/ProxyXmlStartTag.java	2003-04-07 22:51:31.000000000 +0200
+++ ProxyXmlStartTag.java	2007-02-18 00:31:41.034648139 +0100
@@ -35,7 +35,18 @@
     /** The factory used to create new elements */
     private DocumentFactory factory = DocumentFactory.getInstance();
 
-    
+    	/* to compile */
+	public  boolean removeAttributeByName(String a,String b){
+		return true;
+	}
+
+	public  boolean removeAttributeByRawName(String a){
+		return true;
+	}
+
+	public void removeAttributes(){}
+	/**/
+
     public ProxyXmlStartTag() { 
     }
     
