--- /usr/local/tmp/portage/dev-java/dom4j-1.4/work/dom4j-1.4/src/java/org/dom4j/xpath/XPathPattern.java	2003-04-07 22:51:31.000000000 +0200
+++ XPathPattern.java	2007-02-18 01:50:07.177677215 +0100
@@ -57,15 +57,15 @@
         this.context = new Context( getContextSupport() );
     }
 
-    public XPathPattern(String text) {
+    public XPathPattern(String text) throws org.jaxen.JaxenException {
         this.text = text;
         this.context = new Context( getContextSupport() );
         try {
             this.pattern = PatternParser.parse( text );
         }
-        catch (SAXPathException e) {
-            throw new InvalidXPathException( text, e.getMessage() );
-        }
+        catch (org.jaxen.saxpath.SAXPathException e) {
+	            throw new InvalidXPathException( text, e.getMessage() );
+	       }
         catch (RuntimeException e) {
             throw new InvalidXPathException( text );
         }
