diff -ru jasperreports-0.6.8/src/net/sf/jasperreports/engine/design/JRJdtCompiler.java jasperreports-0.6.8-patched/src/net/sf/jasperreports/engine/design/JRJdtCompiler.java
--- jasperreports-0.6.8/src/net/sf/jasperreports/engine/design/JRJdtCompiler.java	2005-05-03 11:37:40.000000000 -0500
+++ jasperreports-0.6.8-patched/src/net/sf/jasperreports/engine/design/JRJdtCompiler.java	2006-02-25 16:55:14.000000000 -0500
@@ -223,7 +223,7 @@
 						ICompilationUnit compilationUnit = 
 							new CompilationUnit(
 								sourceCode, className);
-						return new NameEnvironmentAnswer(compilationUnit);
+						return new NameEnvironmentAnswer(compilationUnit, null);
 					}
 					String resourceName = className.replace('.', '/') + ".class";
 					InputStream is = classLoader.getResourceAsStream(resourceName);
@@ -242,7 +242,7 @@
 						char[] fileName = className.toCharArray();
 						ClassFileReader classFileReader = 
 							new ClassFileReader(classBytes, fileName, true);
-						return new NameEnvironmentAnswer(classFileReader);
+						return new NameEnvironmentAnswer(classFileReader, null);
 					}
 				}
 				catch (IOException exc) 
@@ -440,4 +440,4 @@
 	}
 
 
-}
\ No newline at end of file
+}
