diff -Nur dbus-java-2.4/org/freedesktop/dbus/bin/CreateInterface.java dbus-java-2.4_patched/org/freedesktop/dbus/bin/CreateInterface.java
--- dbus-java-2.4/org/freedesktop/dbus/bin/CreateInterface.java	2008-02-05 11:50:23.000000000 +0200
+++ dbus-java-2.4_patched/org/freedesktop/dbus/bin/CreateInterface.java	2008-06-01 10:46:41.000000000 +0300
@@ -38,6 +38,7 @@
 
 import org.freedesktop.DBus.Introspectable;
 import org.freedesktop.dbus.DBusConnection;
+import org.freedesktop.dbus.Marshalling;
 import org.freedesktop.dbus.exceptions.DBusException;
 import org.freedesktop.dbus.exceptions.DBusExecutionException;
 import org.freedesktop.dbus.types.DBusStructType;
@@ -103,7 +104,7 @@
    {
       if (null == dbus || "".equals(dbus)) return "";
       Vector<Type> v = new Vector<Type>();
-      /* UNNECCESSARY?? int c = Marshalling.getJavaType(dbus, v, 1);*/
+      int c = Marshalling.getJavaType(dbus, v, 1);
       Type t = v.get(0);
       return collapseType(t, imports, structs, container, fullnames);
    }
