--- src/com/mysql/jdbc/ConnectionProperties.java.orig	2005-07-10 20:42:46.000000000 +0200
+++ src/com/mysql/jdbc/ConnectionProperties.java	2005-07-10 20:43:16.000000000 +0200
@@ -26,7 +26,6 @@
 
 import com.mysql.jdbc.log.Jdk14Logger;
 import com.mysql.jdbc.log.Log;
-import com.mysql.jdbc.log.Log4JLogger;
 import com.mysql.jdbc.log.StandardLogger;
 
 import java.io.UnsupportedEncodingException;
@@ -2392,7 +2391,7 @@
 				try {
 					// Is Log4J available?
 					Class.forName("org.apache.log4j.Level");
-					setLogger(Log4JLogger.class.getName());
+					setLogger("com.mysql.jdbc.log.Log4JLogger");
 				} catch (Throwable t) {
 					try {
 						// Are we running on JDK-1.4?
