--- jruby-1.1.6/src/org/jruby/RubyRegexp.java.orig	2008-12-17 19:46:47.000000000 +0000
+++ jruby-1.1.6/src/org/jruby/RubyRegexp.java	2009-04-19 13:02:43.527938812 +0100
@@ -742,7 +742,18 @@
     final int adjustStartPos(RubyString str, int pos, boolean reverse) {
         check();
         ByteList value = str.getByteList();
-        return pattern.adjustStartPosition(value.bytes, value.begin, value.realSize, pos, reverse);
+        Encoding enc = pattern.getEncoding();
+        int len = value.realSize;
+        if (pos > 0 && enc.maxLength() != 1 && pos < len) {
+            int start = value.begin;
+            if ((reverse ? -pos : len - pos) > 0) {
+                return enc.rightAdjustCharHead(value.bytes, start, start + pos, start + len) - start;
+            } else {
+                return enc.leftAdjustCharHead(value.bytes, start, start + pos, start + len) - start;
+            }
+        }
+
+        return pos;
     }
 
     @JRubyMethod(name = "casefold?")
