diff -Nur jwordsplitter-2.1/src/de/abelssoft/wordtools/jWordSplitter/impl/GermanWordSplitter.java jwordsplitter-2.1_patched/src/de/abelssoft/wordtools/jWordSplitter/impl/GermanWordSplitter.java
--- jwordsplitter-2.1/src/de/abelssoft/wordtools/jWordSplitter/impl/GermanWordSplitter.java	2008-07-02 00:08:53.000000000 +0300
+++ jwordsplitter-2.1_patched/src/de/abelssoft/wordtools/jWordSplitter/impl/GermanWordSplitter.java	2008-10-26 20:24:22.000000000 +0200
@@ -33,6 +33,8 @@
   private static final String SERIALIZED_DICT = "/wordsGerman.ser";   // dict inside the JAR
 
   private HashSet words = null;
+
+  private int minimumWordLength = 4;
 	
 	public GermanWordSplitter() throws IOException
 	{
@@ -44,10 +46,10 @@
 		super(hideConnectingCharacters, plainTextDictFile);
 	}
 
-	 public GermanWordSplitter(boolean hideConnectingCharacters) throws IOException
-	  {
-	    super(hideConnectingCharacters);
-	  }
+	public GermanWordSplitter(boolean hideConnectingCharacters) throws IOException
+	{
+	  super(hideConnectingCharacters);
+	}
 
 	protected HashSet getWordList() throws IOException
 	{
@@ -68,9 +70,14 @@
 	
 	protected int getMinimumWordLength()
 	{
-		return 4;
+		return minimumWordLength ;
 	}
 
+	public void setMinimumWordLength(int minimumWordLength)
+  {
+    this.minimumWordLength = minimumWordLength;
+  }
+
 	protected String getConnectingCharacters()
 	{
 		return "s";
