diff -u -r1.3 LibGlade.java.in
--- libglade-java/src/java/org/gnu/glade/LibGlade.java.in	24 Oct 2004 12:19:55 -0000	1.3
+++ libglade-java/src/java/org/gnu/glade/LibGlade.java.in	24 Dec 2004 00:39:38 -0000
@@ -88,8 +88,9 @@
 		/* When you call glade_xml_new_from_buffer, it looses
 		 * information about what the glade file was, and later runs
 		 * into issues if you try to instantiate anything that needs a
-		 * path relative to that file for example, a GtkPibuf */
+		 * path relative to that file for example, a GtkPixbuf */
 
+		this.owner = owner;
 		glade_xml_new(file, root);
 		glade_xml_signal_autoconnect_full();
 	}
