--- Makefile.am.orig	2005-11-12 14:32:39.000000000 -0600
+++ Makefile.am	2005-11-12 14:33:53.000000000 -0600
@@ -985,8 +985,8 @@
 pkgconfig_DATA = gtk2-java.pc
 
 # install the macros used by other java-gnome packages
-jgmacrodir = $(datadir)/libgtk-java/macros
-jgmacro_DATA = \
+m4datadir = $(datadir)/aclocal
+m4data_DATA = \
 	macros/jg_gtk_java.m4 \
 	macros/jg_gnome_java.m4
 
--- gtk2-java.pc.in.orig	2005-11-12 14:34:13.000000000 -0600
+++ gtk2-java.pc.in	2005-11-12 14:34:25.000000000 -0600
@@ -10,7 +10,6 @@
 Version: @VERSION@
 api_version=@gtkapiversion@
 classpath=@INSTALLED_CLASSPATH@
-macro_dir=@INSTALLED_MACRO_DIR@
 jnilibs=-L@libdir@ -lgtkjni
 Libs: -L@libdir@ -lgtkjava
 Cflags: -I@includedir@/libgtk-java
