diff -Nuar libgtk-java-2.4.8.1-ORIG/Makefile.in libgtk-java-2.4.8.1-WORK/Makefile.in
--- libgtk-java-2.4.8.1-ORIG/Makefile.in	2005-02-12 11:52:04.000000000 +1100
+++ libgtk-java-2.4.8.1-WORK/Makefile.in	2005-02-26 20:18:44.000000000 +1100
@@ -1407,7 +1407,7 @@
 @BUILD_GCJ_TRUE@am_libgtkjni_la_rpath = -rpath $(libdir)
 exampleSCRIPT_INSTALL = $(INSTALL_SCRIPT)
 SCRIPTS = $(example_SCRIPTS)
-DEFAULT_INCLUDES = -I. -I$(srcdir)
+DEFAULT_INCLUDES = -I. -I$(srcdir) -I${JDK_HOME}/include -I${JDK_HOME}/include/linux/
 depcomp = $(SHELL) $(top_srcdir)/depcomp
 am__depfiles_maybe = depfiles
 COMPILE = $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) \
