diff -Nuar libgtk-java-2.4.6-ORIG/Makefile.in libgtk-java-2.4.6-EDIT/Makefile.in
--- libgtk-java-2.4.6-ORIG/Makefile.in	2004-10-07 04:54:44.000000000 +1000
+++ libgtk-java-2.4.6-EDIT/Makefile.in	2004-11-08 16:51:29.352689812 +1100
@@ -37,7 +37,7 @@
 # Install core gtk java libraries and required libraries
 install:  @GCJ_INSTALL_TARGETS@
 	-$(INSTALL) -d $(DESTDIR)$(datadir)/java/
-	-$(INSTALL) -d $(DESTDIR)$(datadir)/doc/libgtk$(apiversion)-java/
+	-$(INSTALL) -d $(DESTDIR)$(datadir)/doc/libgtk$(gtkapiversion)-java/
 	-$(INSTALL) -d $(DESTDIR)$(libdir)/
 	$(INSTALL) -s -m644 lib/libgtkjava$(gtkapiversion).so.$(gtkversion) $(DESTDIR)$(libdir)/
 	$(INSTALL) -m644 lib/gtk$(gtkapiversion)-$(gtkversion).jar $(DESTDIR)$(datadir)/java/
