--- libgtk-java-2.4.8.1-ORIG/configure	2005-02-12 11:52:07.000000000 +1100
+++ libgtk-java-2.4.8.1-WORK/configure	2005-02-24 15:09:31.000000000 +1100
@@ -19806,9 +19806,9 @@
 gtkapiversion="2.4"
 
 
-INSTALLED_CLASSPATH=\${prefix}/share/java/gtk${gtkapiversion}-${VERSION}.jar
+INSTALLED_CLASSPATH=\${prefix}/share/${PN}-${SLOT}/lib/gtk${gtkapiversion}-${VERSION}.jar
 INSTALLED_JNI_DIR=\${prefix}/lib/
-INSTALLED_MACRO_DIR=\${prefix}/share/libgtk-java/macros/
+INSTALLED_MACRO_DIR=\${prefix}/share/${PN}-${SLOT}/macros/
 
 
 
