--- libgtk-java-2.4.8.1/src/jni/org_gnu_gtk_TextBuffer.c	2005-01-31 13:15:32.000000000 +1100
+++ cvs/gnome2/java-gnome-2.8/libgtk-java/src/jni/org_gnu_gtk_TextBuffer.c	2005-02-25 02:57:45.000000000 +1100
@@ -68,11 +68,11 @@
  * Method:    gtk_text_buffer_get_tag_table
  * Signature: (I)I
  */
-JNIEXPORT jint JNICALL Java_org_gnu_gtk_TextBuffer_gtk_1text_1buffer_1get_1tag_1table (JNIEnv *env, jclass cls, jobject buffer) 
+JNIEXPORT jobject JNICALL Java_org_gnu_gtk_TextBuffer_gtk_1text_1buffer_1get_1tag_1table (JNIEnv *env, jclass cls, jobject buffer) 
 {
     GtkTextBuffer *buffer_g = 
         (GtkTextBuffer*)getPointerFromHandle(env, buffer);
-    return (jint)gtk_text_buffer_get_tag_table (buffer_g);
+    return getHandleFromPointer(env, gtk_text_buffer_get_tag_table (buffer_g));
 }
 
 /*
