--- CsvBuilder.java	2008-02-12 21:19:40.000000000 +0200
+++ CsvBuilder.java.fixed	2008-03-11 20:51:32.000000000 +0200
@@ -33,9 +33,6 @@
 
 import com.csvreader.CsvReader;
 import com.csvreader.CsvWriter;
-import com.csvreader.CsvReader.CatastrophicException;
-import com.csvreader.CsvReader.FinalizedException;
-
 /**
  * A class for importing and exporting character-separated-values (CSV).
  * 
@@ -66,8 +63,6 @@
     public static void exportLesson(OutputStream out, Lesson lesson, 
         char delimiter, Charset charset) throws IOException
     {
-        try
-        {
             CsvWriter writer = new CsvWriter(out, delimiter, charset);
             writeHeader(writer);
             
@@ -87,11 +82,6 @@
             }
             
             writer.close();
-        }
-        catch (com.csvreader.CsvWriter.FinalizedException e)
-        {
-            throw new IOException(e.getMessage());
-        }
     }
     
     
@@ -123,8 +113,6 @@
             categories.put(category.getName(), category);
         }
         
-        try
-        {
             reader.readHeaders();
             
             String[] headers = reader.getHeaders();
@@ -174,19 +162,9 @@
             }
 
             reader.close();
-        } 
-        catch (FinalizedException e)
-        {
-            throw new IOException(e.toString());
-        } 
-        catch (CatastrophicException e)
-        {
-            throw new IOException(e.toString());
-        }
     }
 
-    private static void writeHeader(CsvWriter writer) throws IOException, 
-        com.csvreader.CsvWriter.FinalizedException
+    private static void writeHeader(CsvWriter writer) throws IOException
     {
         writer.write(FRONTSIDE_COL);
         writer.write(FLISIDE_COL);
