Index: src/org/jruby/ast/visitor/AbstractVisitor.java
===================================================================
--- src/org/jruby/ast/visitor/AbstractVisitor.java	(revision 8497)
+++ src/org/jruby/ast/visitor/AbstractVisitor.java	(working copy)
@@ -85,6 +85,7 @@
 import org.jruby.ast.Match3Node;
 import org.jruby.ast.MatchNode;
 import org.jruby.ast.ModuleNode;
+import org.jruby.ast.MultipleAsgn19Node;
 import org.jruby.ast.MultipleAsgnNode;
 import org.jruby.ast.NewlineNode;
 import org.jruby.ast.NextNode;
@@ -103,6 +104,7 @@
 import org.jruby.ast.RegexpNode;
 import org.jruby.ast.RescueBodyNode;
 import org.jruby.ast.RescueNode;
+import org.jruby.ast.RestArgNode;
 import org.jruby.ast.RetryNode;
 import org.jruby.ast.ReturnNode;
 import org.jruby.ast.RootNode;
@@ -525,4 +527,12 @@
     public Instruction visitAttrAssignNode(AttrAssignNode iVisited) {
         return visitNode(iVisited);
     }
+    
+    public Instruction visitRestArgNode(RestArgNode iVisited) {
+        return visitNode(iVisited);
+    }
+    
+    public Instruction visitMultipleAsgnNode(MultipleAsgn19Node iVisited) {
+        return visitNode(iVisited);
+    }
 }
