--- libgnome-java-2.10.0/Makefile.in	2005-03-06 18:14:01.000000000 -0500
+++ fixed/Makefile.in	2005-04-04 00:39:40.000000000 -0400
@@ -3072,7 +3072,7 @@
 	$(DOCBOOK) -o doc/doc $<
 
 install-data-hook:
-	cd $(DESTDIR)$(datadir)/java && \
+	cd $(DESTDIR)$(jardir) && \
 		mv -f gnome$(apiversion).jar gnome$(apiversion)-$(VERSION).jar && \
 		$(LN_S) gnome$(apiversion)-$(VERSION).jar gnome$(apiversion).jar
 
