Index: configure.ac
===================================================================
RCS file: /cvsroot/classpath/classpath/configure.ac,v
retrieving revision 1.118
diff -u -r1.118 configure.ac
--- configure.ac	7 Nov 2005 14:25:54 -0000	1.118
+++ configure.ac	10 Nov 2005 10:50:23 -0000
@@ -118,7 +118,7 @@
 dnl -----------------------------------------------------------
 dnl DSSI code (enabled by default)
 dnl -----------------------------------------------------------
-AC_ARG_ENABLE([alsa],
+AC_ARG_ENABLE([dssi],
               [AS_HELP_STRING(--disable-dssi,compile DSSI providers (enable by --enable-dssi) [default=yes])],
               [case "${enableval}" in
                 yes) COMPILE_DSSI=yes ;;
Index: native/jni/midi-dssi/Makefile.am
===================================================================
RCS file: /cvsroot/classpath/classpath/native/jni/midi-dssi/Makefile.am,v
retrieving revision 1.2
diff -u -r1.2 Makefile.am
--- native/jni/midi-dssi/Makefile.am	23 Oct 2005 16:59:09 -0000	1.2
+++ native/jni/midi-dssi/Makefile.am	10 Nov 2005 10:50:23 -0000
@@ -1,7 +1,8 @@
 nativelib_LTLIBRARIES = libgjsmdssi.la 
 
 libgjsmdssi_la_SOURCES = gnu_javax_sound_midi_dssi_DSSIMidiDeviceProvider.c \
-gnu_javax_sound_midi_dssi_DSSISynthesizer.c
+			 gnu_javax_sound_midi_dssi_DSSISynthesizer.c \
+			 dssi_data.h
 
 libgjsmdssi_la_LIBADD = $(top_builddir)/native/jni/classpath/jcl.lo -ljack
 
