--- classpath/classpath/native/jni/gtk-peer/Makefile.am	2005/08/22 10:36:141.29
+++ classpath/classpath/native/jni/gtk-peer/Makefile.am	2005/10/21 19:26:531.30
@@ -54,10 +54,12 @@
 libgtkpeer_la_LIBADD = $(top_builddir)/native/jni/classpath/native_state.lo \
        $(top_builddir)/native/jni/classpath/jcl.lo
 
-AM_LDFLAGS = @CLASSPATH_MODULE@ @GTK_LIBS@ @CAIRO_LIBS@ @PANGOFT2_LIBS@ @X_LIBS@ -lXtst
+AM_LDFLAGS = @CLASSPATH_MODULE@ @GTK_LIBS@ @CAIRO_LIBS@ @PANGOFT2_LIBS@ \
+             @FREETYPE2_LIBS@ @X_PRE_LIBS@ @X_LIBS@ @X_EXTRA_LIBS@ -lX11 -lXtst
 AM_CPPFLAGS = @CLASSPATH_INCLUDES@
 
 # Just the WARNING_CFLAGS. We cannot use the strict flags since the gtk
 # headers contain broken prototypes (by design, see gtkitemfactory.h).
 AM_CFLAGS = @WARNING_CFLAGS@ @ERROR_CFLAGS@ \
-            @GTK_CFLAGS@ @CAIRO_CFLAGS@ @PANGOFT2_CFLAGS@
+            @GTK_CFLAGS@ @CAIRO_CFLAGS@ @PANGOFT2_CFLAGS@ @FREETYPE2_CFLAGS@ \
+            @X_CFLAGS@

--- classpath/classpath/configure.ac	2005/10/06 06:24:111.111
+++ classpath/classpath/configure.ac	2005/10/21 19:26:531.112
@@ -373,12 +373,17 @@
 
   if test "x${enable_gtk_cairo}" = xyes; then
     PKG_CHECK_MODULES(CAIRO, cairo >= 0.5.0)
-    PKG_CHECK_MODULES(PANGOFT2, pangoft2)
   fi
+
+  PKG_CHECK_MODULES(PANGOFT2, pangoft2)
+  PKG_CHECK_MODULES(FREETYPE2, freetype2)
+
   AC_SUBST(CAIRO_LIBS)
   AC_SUBST(CAIRO_CFLAGS)
   AC_SUBST(PANGOFT2_LIBS)
   AC_SUBST(PANGOFT2_CFLAGS)
+  AC_SUBST(FREETYPE2_LIBS)
+  AC_SUBST(FREETYPE2_CFLAGS)
 fi
 
 CLASSPATH_WITH_JAVAH
