--- Makefile.am.orig	2005-11-11 22:34:15.000000000 -0600
+++ Makefile.am	2005-11-11 22:36:54.000000000 -0600
@@ -53,8 +53,8 @@
 pkgconfig_DATA = glib-java.pc
 
 # install the macros used by other java-gnome packages
-glibmacrodir = $(datadir)/glib-java/macros
-glibmacro_DATA = \
+m4datadir = $(datadir)/aclocal
+m4data_DATA= \
 	macros/jg_common.m4 \
 	macros/jg_check_nativecompile.m4 \
 	macros/am_path_gcj.m4 \
--- glib-java.pc.in  2005-11-11 22:29:40.000000000 -0600
+++ glib-java.pc.in	2005-11-11 22:29:52.000000000 -0600
@@ -9,7 +9,6 @@
 Version: @VERSION@
 api_version=@apiversion@
 classpath=@INSTALLED_CLASSPATH@
-macro_dir=@INSTALLED_MACRO_DIR@
 jnilibs=-L@libdir@ -lglibjni
 Libs: -L@libdir@ -lglibjava
 Cflags: -I@includedir@/glib-java
