diff -ru glib-java-0.2.5-orig/macros/am_path_gcj.m4 glib-java-0.2.5/macros/am_path_gcj.m4
--- glib-java-0.2.5-orig/macros/am_path_gcj.m4	2006-06-20 08:42:59.000000000 -0500
+++ glib-java-0.2.5/macros/am_path_gcj.m4	2006-06-20 08:43:20.000000000 -0500
@@ -53,7 +53,7 @@
   if test "$GCJ" = "no" ; then
     no_gcj=yes
   else
-    gcj_version=`$GCJ --version 2>&1 | grep GCC | sed 's/gcj.* (GCC) \([[0-9]]*\)\.\([[0-9]]*\)\.\([[0-9]]*\).*/\1.\2.\3/'`
+    gcj_version=`$GCJ -dumpversion`
     gcj_major_version=`echo $gcj_version | \
            sed 's/\([[0-9]]*\)\.\([[0-9]]*\)\.\([[0-9]]*\).*/\1/'`
     gcj_minor_version=`echo $gcj_version | \
