--- macros/jg_common.m4	2005-04-13 10:35:40.000000000 -0500
+++ macros/jg_common.m4	2005-11-11 00:05:15.000000000 -0600
@@ -4,7 +4,7 @@
 
 dnl check to see if the user wants to generate javadoc
 AC_ARG_WITH(javadocs,[  --without-javadocs    Do not build the javadocs for the bindings],
-        javadocs="no", javadocs="yes")
+        javadocs=$withval, javadocs="yes")
 AM_CONDITIONAL(BUILD_JAVADOC, test $javadocs = "yes")
 
 AC_ARG_WITH([jardir], AS_HELP_STRING([--with-jardir],
