diff -Nru resin-3.0.15.vanilla/modules/resin/src/com/caucho/server/resin/Resin.java resin-3.0.15/modules/resin/src/com/caucho/server/resin/Resin.java
--- resin-3.0.15.vanilla/modules/resin/src/com/caucho/server/resin/Resin.java	2005-11-14 19:25:57.000000000 +0100
+++ resin-3.0.15/modules/resin/src/com/caucho/server/resin/Resin.java	2005-11-25 17:06:42.000000000 +0100
@@ -689,10 +689,10 @@
       throw new ConfigException(L().l("The following system property must be set:\n  -Djava.util.logging.manager=com.caucho.log.LogManagerImpl\nThe JDK 1.4 Logging manager must be set to Resin's log manager."));
     }
 
-    validatePackage("javax.servlet.Servlet", new String[] {"2.4", "1.4"});
+    validatePackage("javax.servlet.Servlet", new String[] {"2.4", "1.5"});
     validatePackage("javax.servlet.jsp.jstl.core.Config", new String[] {"1.1"});
-    validatePackage("javax.management.MBeanServer", new String[] {"1.2" });
-    validatePackage("javax.resource.spi.ResourceAdapter", new String[] {"1.5", "1.4"});
+    validatePackage("javax.management.MBeanServer", new String[] {"1.2", "1.5" });
+    validatePackage("javax.resource.spi.ResourceAdapter", new String[] {"1.5", "1.5"});
   }
 
   /**
