--- build.xml	2007-01-17 22:21:10.000000000 +0200
+++ build.xml.fixed	2007-05-16 21:44:55.000000000 +0300
@@ -141,6 +141,22 @@
       <echo message="Excluding: ${excludes}"/>
       <javac classpath="${classpath}" debug="on" deprecation="yes" destdir="${lbuild}" excludes="${excludes}" includes="**/*.java" srcdir="${src}"> </javac>
   </target>
+  
+  <target depends="build" name="jar-engine">
+    <jar jarfile="joone-engine.jar">
+        <fileset dir="${build}" includes="${joone_eng_engine}"/>
+        <fileset dir="${build}" includes="${joone_eng_io}"/>
+        <fileset dir="${build}" includes="${joone_eng_util}"/>
+        <fileset dir="${build}" includes="${joone_eng_struc}"/>
+        <fileset dir="${build}" includes="${joone_eng_helpers}"/>
+        <fileset dir="${build}" includes="${joone_eng_log}"/>
+        <fileset dir="${build}" includes="${joone_eng_net}"/>
+        <fileset dir="${build}" includes="${joone_eng_script}"/>
+        <fileset dir="${build}" includes="${joone_eng_inspection}"/>
+        <fileset dir="${build}" includes="${joone_eng_exception}"/>
+        <fileset dir="${build}" includes="${joone_eng_samples}"/>
+    </jar>
+  </target>
 
   <target depends="init" name="release">
     <delete>
