Author: Ralph Sennhauser <sera@gentoo.org>

Disables -Werror so shark builds with llvm-3.0 on amd64

--- openjdk/hotspot/make/linux/makefiles/gcc.make	2011-12-12 09:38:55.995256804 +0100
+++ openjdk/hotspot/make/linux/makefiles/gcc.make	2011-12-12 10:26:07.670560007 +0100
@@ -123,7 +123,7 @@
 endif
 
 # Compiler warnings are treated as errors
-WARNINGS_ARE_ERRORS = -Werror
+WARNINGS_ARE_ERRORS =
 
 # Except for a few acceptable ones
 # Since GCC 4.3, -Wconversion has changed its meanings to warn these implicit
