diff -r afb4ea482b75 Makefile.am
--- a/Makefile.am	Sun Apr 13 00:46:37 2008 +0200
+++ b/Makefile.am	Mon Apr 14 02:38:24 2008 +0100
@@ -97,7 +97,9 @@ ICEDTEA_ENV = \
 	"LIBFFI_CFLAGS=$(LIBFFI_CFLAGS)" \
 	"LIBFFI_LIBS=$(LIBFFI_LIBS)" \
 	"FREETYPE2_HEADERS=$(FREETYPE2_CFLAGS)" \
-	"FT2_LIB=$(FREETYPE2_LIBS)"
+	"FT2_LIB=$(FREETYPE2_LIBS)" \
+	"ALT_PARALLEL_COMPILE_JOBS=$(PARALLEL_JOBS)" \
+	"HOTSPOT_BUILD_JOBS=$(PARALLEL_JOBS)"
 
 if WITH_CACAO
 ICEDTEA_ENV += \
@@ -152,7 +154,9 @@ ICEDTEA_ENV_ECJ = \
 	"LIBFFI_CFLAGS=$(LIBFFI_CFLAGS)" \
 	"LIBFFI_LIBS=$(LIBFFI_LIBS)" \
 	"FREETYPE2_HEADERS=$(FREETYPE2_CFLAGS)" \
-	"FT2_LIB=$(FREETYPE2_LIBS)"
+	"FT2_LIB=$(FREETYPE2_LIBS)" \
+	"ALT_PARALLEL_COMPILE_JOBS=$(PARALLEL_JOBS)" \
+	"HOTSPOT_BUILD_JOBS=$(PARALLEL_JOBS)"
 
 if WITH_CACAO
 ICEDTEA_ENV_ECJ += \
diff -r afb4ea482b75 Makefile.in
--- a/Makefile.in	Sun Apr 13 00:46:37 2008 +0200
+++ b/Makefile.in	Mon Apr 14 02:38:24 2008 +0100
@@ -150,6 +150,7 @@ PACKAGE_STRING = @PACKAGE_STRING@
 PACKAGE_STRING = @PACKAGE_STRING@
 PACKAGE_TARNAME = @PACKAGE_TARNAME@
 PACKAGE_VERSION = @PACKAGE_VERSION@
+PARALLEL_JOBS = @PARALLEL_JOBS@
 PATCH = @PATCH@
 PATH_SEPARATOR = @PATH_SEPARATOR@
 PKG_CONFIG = @PKG_CONFIG@
@@ -295,7 +296,10 @@ ICEDTEA_ENV = "ALT_JDK_IMPORT_PATH=$(ICE
 	"ZERO_ARCHFLAG=$(ZERO_ARCHFLAG)" \
 	"LIBFFI_CFLAGS=$(LIBFFI_CFLAGS)" "LIBFFI_LIBS=$(LIBFFI_LIBS)" \
 	"FREETYPE2_HEADERS=$(FREETYPE2_CFLAGS)" \
-	"FT2_LIB=$(FREETYPE2_LIBS)" $(am__append_1) $(am__append_2)
+	"FT2_LIB=$(FREETYPE2_LIBS)" \
+	"ALT_PARALLEL_COMPILE_JOBS=$(PARALLEL_JOBS)" \
+	"HOTSPOT_BUILD_JOBS=$(PARALLEL_JOBS)" $(am__append_1) \
+	$(am__append_2)
 
 # OpenJDK ecj build environment.
 ICEDTEA_BUILD_DIR_ECJ = \
@@ -332,7 +336,10 @@ ICEDTEA_ENV_ECJ = "ALT_JDK_IMPORT_PATH=$
 	"ZERO_ARCHFLAG=$(ZERO_ARCHFLAG)" \
 	"LIBFFI_CFLAGS=$(LIBFFI_CFLAGS)" "LIBFFI_LIBS=$(LIBFFI_LIBS)" \
 	"FREETYPE2_HEADERS=$(FREETYPE2_CFLAGS)" \
-	"FT2_LIB=$(FREETYPE2_LIBS)" $(am__append_3) $(am__append_4)
+	"FT2_LIB=$(FREETYPE2_LIBS)" \
+	"ALT_PARALLEL_COMPILE_JOBS=$(PARALLEL_JOBS)" \
+	"HOTSPOT_BUILD_JOBS=$(PARALLEL_JOBS)" $(am__append_3) \
+	$(am__append_4)
 
 # OpenJDK Source Preparation Targets
 # ==================================
diff -r afb4ea482b75 configure
--- a/configure	Sun Apr 13 00:46:37 2008 +0200
+++ b/configure	Mon Apr 14 02:38:24 2008 +0100
@@ -728,6 +728,7 @@ WITH_ICEDTEA_FALSE
 WITH_ICEDTEA_FALSE
 WITH_OPENJDK_TRUE
 WITH_OPENJDK_FALSE
+PARALLEL_JOBS
 BUILD_ARCH_DIR
 INSTALL_ARCH_DIR
 JAVA
@@ -1438,6 +1439,8 @@ Optional Packages:
   --with-ant-home         Ant home directory (default is /usr/share/ant)
   --with-icedtea          build IcedTea with system-installed IcedTea
   --with-openjdk          build IcedTea with system-installed OpenJDK
+  --with-parallel-jobs    build IcedTea using the specified number of parallel
+                          jobs
   --with-java             specify location of the 1.5 java vm
   --with-ecj              bytecode compilation with ecj
   --with-javac            bytecode compilation with javac
@@ -5608,6 +5611,25 @@ else
   WITH_OPENJDK_TRUE='#'
   WITH_OPENJDK_FALSE=
 fi
+
+
+
+# Check whether --with-parallel-jobs was given.
+if test "${with_parallel_jobs+set}" = set; then
+  withval=$with_parallel_jobs;
+          if test "x${withval}" == x
+          then
+            PARALLEL_JOBS=2
+          else
+            PARALLEL_JOBS=${withval}
+          fi
+
+else
+
+          PARALLEL_JOBS=2
+
+fi
+
 
 
 
@@ -12592,6 +12614,7 @@ WITH_ICEDTEA_FALSE!$WITH_ICEDTEA_FALSE$a
 WITH_ICEDTEA_FALSE!$WITH_ICEDTEA_FALSE$ac_delim
 WITH_OPENJDK_TRUE!$WITH_OPENJDK_TRUE$ac_delim
 WITH_OPENJDK_FALSE!$WITH_OPENJDK_FALSE$ac_delim
+PARALLEL_JOBS!$PARALLEL_JOBS$ac_delim
 BUILD_ARCH_DIR!$BUILD_ARCH_DIR$ac_delim
 INSTALL_ARCH_DIR!$INSTALL_ARCH_DIR$ac_delim
 JAVA!$JAVA$ac_delim
@@ -12663,7 +12686,7 @@ LTLIBOBJS!$LTLIBOBJS$ac_delim
 LTLIBOBJS!$LTLIBOBJS$ac_delim
 _ACEOF
 
-  if test `sed -n "s/.*$ac_delim\$/X/p" conf$$subs.sed | grep -c X` = 85; then
+  if test `sed -n "s/.*$ac_delim\$/X/p" conf$$subs.sed | grep -c X` = 86; then
     break
   elif $ac_last_try; then
     { { echo "$as_me:$LINENO: error: could not make $CONFIG_STATUS" >&5
diff -r afb4ea482b75 configure.ac
--- a/configure.ac	Sun Apr 13 00:46:37 2008 +0200
+++ b/configure.ac	Mon Apr 14 02:38:24 2008 +0100
@@ -139,6 +139,22 @@ AC_ARG_WITH([openjdk],
         ])
 AM_CONDITIONAL(WITH_OPENJDK, test "${with_openjdk}" == true)
 
+AC_ARG_WITH([parallel-jobs],
+	[AS_HELP_STRING([--with-parallel-jobs],
+			[build IcedTea using the specified number of parallel jobs])],
+	[
+          if test "x${withval}" == x
+          then
+            PARALLEL_JOBS=2
+          else
+            PARALLEL_JOBS=${withval}
+          fi
+        ],
+        [
+          PARALLEL_JOBS=2
+        ])
+AC_SUBST(PARALLEL_JOBS)
+	
 SET_ARCH_DIRS
 if test "${with_openjdk}" == true
 then
