diff -r 0883b7a8311a acinclude.m4
--- a/acinclude.m4	Fri Jan 29 22:47:27 2016 +0000
+++ b/acinclude.m4	Fri Jan 29 23:24:48 2016 +0000
@@ -1969,7 +1969,7 @@
     PKG_CHECK_MODULES(NSS_JAVA, nss-java, [NSS_JAVA_FOUND=yes], [NSS_JAVA_FOUND=no])
     if test "x${NSS_SOFTOKN_FOUND}" = "xyes"; then
       SUNEC_CFLAGS=$NSS_SOFTOKN_CFLAGS;
-      SUNEC_LIBS="$NSS_LIBS -lfreebl";
+      SUNEC_LIBS="-lfreebl $NSS_LIBS";
    elif test "x${NSS_JAVA_FOUND}" = "xyes"; then
       SUNEC_CFLAGS="$NSS_JAVA_CFLAGS -DLEGACY_NSS";
       SUNEC_LIBS=$NSS_JAVA_LIBS;
